---
title: "Fig_TFdel"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, warning=FALSE}
sapply(c("dplyr", "readr", "tidyr", "purrr", "ggplot2", "ggpubr", "matrixStats", "ComplexHeatmap", "circlize", "ggVennDiagram", "ggExtra"), require, character.only=TRUE)
load("data_files/CleanedCounts.RData")
load("data_files/DESeq2.RData")
load("data_files/GeneDataFrame.RData")
source("utils.R")

sgd_lookup <- read_delim("data_files/downloaded_genomes_and_features/SGD_features.tab", delim = "\t", col_names = FALSE) |> filter(X2 == "ORF") |> 
  select(X4, X5) |> 
  dplyr::rename("gene_name"="X4",
                "common_name"="X5")

FullTFnames <- setdiff(unique(gsub("delete", "", sample_info$genotype)), "WT")
NitrogenTFs <- c("URE2", "GAT1", "GLN3", "DAL80", "GZF3")
# ordering TFnames so that nitrogen comes first then alphabetical
TFnames <- c(NitrogenTFs, sort(setdiff(FullTFnames, NitrogenTFs)))

p_thresh <- 0.05 # because DESeq2 already corrected for FDR with alpha = 0.05
eff_thresh <- 2
TFdel_lookup <- read_delim("data_files/downloaded_genomes_and_features/yeastract_46TFs.csv", col_names = FALSE, col_select = c(1,2), delim = ";") # gets some warnings, but so far has been fine
colnames(TFdel_lookup) <- c("common", "systematic")
```

## Which TFs are differentially expressed between species?

Even though this is only using WT data, this is more relevant to the TFdels, so we'll start with this. Mean expression in Scer vs Spar, averaged across timepoints and replicates. Error bars are standard error from replicates.

```{r}
# scatter plot Scer vs Spar with geom_text as the TF names (b/c the heatmap was only useful when there was more than one environment)
plotdf <- tibble(tf = TFnames) |> 
  left_join(y = sgd_lookup, 
            by = c("tf"="common_name"))
plotdf$mean_expr_cer <- map(plotdf$gene_name, \(g) {
  expr_vec <- NA
  if (g %in% rownames(counts)) {
    expr_vec <- counts[g, sample_info$genotype == "WT" &
                       sample_info$organism == "cer"] |> 
    as.numeric()
  }
  return(mean(expr_vec))
}) |> unlist()
plotdf$mean_expr_par <- map(plotdf$gene_name, \(g) {
  expr_vec <- NA
  if (g %in% rownames(counts)) {
    expr_vec <- counts[g, sample_info$genotype == "WT" &
                       sample_info$organism == "par"] |> 
    as.numeric()
  }
  return(mean(expr_vec))
}) |> unlist()

print(ggplot(plotdf, aes(x = log2(mean_expr_cer + 1), 
                   y = log2(mean_expr_par + 1))) +
  geom_text(aes(label = tf, color = tf %in% NitrogenTFs)) +
  geom_abline(slope = 1, intercept = 0))
```

## How many genes does each TFdel affect in each species?

How many genes are affected in shams vs real knockout lines?
```{r}
plotdf <- bind_rows(mutate(TFdeldf, type = "real"),
                    mutate(TFdeldf_sham1, type = "sham1"),
                    mutate(TFdeldf_sham2, type = "sham2")) |> 
  filter(organism == allele) |> 
  select(gene_name, organism, lfc, padj, type) |> 
  group_by(organism, type) |> 
  summarise(nSig = sum(abs(lfc) > eff_thresh & padj < p_thresh))
p <- ggplot(plotdf, aes(x = type, 
                        y = nSig)) +
  geom_histogram(aes(fill = organism), stat = "identity") +
  geom_text(aes(x = type, y = nSig, label = nSig), nudge_y = 500) +
  scale_fill_discrete(limits = c("cer", "par", "hyb"),
                      type = c(conservation_colordf$value[conservation_colordf$limits == "Scer"],
                               conservation_colordf$value[conservation_colordf$limits == "Spar"],
                               conservation_colordf$value[conservation_colordf$limits == "Hyb"])) +
  facet_wrap(~factor(organism, levels = c("cer", "par", "hyb"),
                     labels = c("Scer", "Spar", "Hyb"))) +
  theme_classic() +
  xlab("") +
  ylab("Number of significantly differentiall expressed\ngenes across all 46 TF knockouts") +
  theme(legend.position = "none")
p
```

For the paper figure:
```{r figure_sham}
# pdf("../Umich_Dissertation/chapters/figures/Networks/shams.pdf",
#     width = 5, height = 4)
# p
# dev.off()
```


Barplots of nGenes up or down in response to each TF deletion.

If we only consider a TF effect shared if it affects the same gene in multiple yeasts and occurs at the same timepoint and direction:
```{r timepoint-specific-effects}
plotdf <- TFdeldf |> filter(padj < p_thresh &
                             abs(lfc) > eff_thresh) |> 
  mutate(lfc_sign = sign(lfc)) |> 
  select(gene_name, deletion, organism, timepoint, lfc_sign) |>
  group_by(gene_name, deletion, timepoint, lfc_sign) |> 
  summarise(orgs = list(organism))
cat("Fraction of single-yeast effects:", sum(sapply(plotdf$orgs, length) == 1), "/", nrow(plotdf),
    "(", sum(sapply(plotdf$orgs, length) == 1)/nrow(plotdf), "%)", "\n")
```
If we instead consider a TF effect shared if it affects the same gene in multiple yeasts in the same direciton, but occurs at any timepoint:
```{r timepoint-agnostic-effects}
plotdf <- TFdeldf |> filter(padj < p_thresh &
                             abs(lfc) > eff_thresh) |> 
  mutate(lfc_sign = sign(lfc)) |> 
  select(gene_name, deletion, organism, lfc_sign) |>
  #unique() |> # unique stops double-counting of gene-deletion-lfc_sign combos that affect multiple timepoints, but that currently results in single-species effects becoming a larger portion of the total effects than the timepoint-specific chunk above, which is counterintuitive so commenting it for now
  group_by(gene_name, deletion, lfc_sign) |> 
  summarise(orgs = list(organism))
cat("Fraction of single-yeast effects:", sum(sapply(plotdf$orgs, length) == 1), "/", nrow(plotdf),
    "(", sum(sapply(plotdf$orgs, length) == 1)/nrow(plotdf), "%)", "\n")
```
That's not a very big difference, so for now we'll specify they're shared if they affect the same TF. Venn diagram makes it even clearer that it doesn't matter so we might as well choose the loosest, simplest definition:

```{r venn}
sigdf <- filter(TFdeldf, padj < p_thresh & abs(lfc) > eff_thresh & organism == allele)
plotdf <- sigdf |> filter(organism == allele)
# TF-gene combo is shared:
p <- ggVennDiagram(list(Scer = paste(plotdf$gene_name[plotdf$organism == "cer"], 
                                plotdf$deletion[plotdf$organism == "cer"], sep = "_"), 
                   Spar = paste(plotdf$gene_name[plotdf$organism == "par"], 
                                plotdf$deletion[plotdf$organism == "par"], sep = "_"),
                   Hyb = paste(plotdf$gene_name[plotdf$organism == "hyb"], 
                                plotdf$deletion[plotdf$organism == "hyb"], sep = "_"))) + 
                scale_fill_gradient(low="grey90",high = "red") +
  ggtitle("TF-gene combination is shared")
# TF-gene-direction combo is shared:
p_dir <- ggVennDiagram(list(Scer = paste(plotdf$gene_name[plotdf$organism == "cer"], 
                                plotdf$deletion[plotdf$organism == "cer"],
                                sign(plotdf$lfc)[plotdf$organism == "cer"], sep = "_"), 
                   Spar = paste(plotdf$gene_name[plotdf$organism == "par"], 
                                plotdf$deletion[plotdf$organism == "par"],
                                sign(plotdf$lfc)[plotdf$organism == "par"], sep = "_"),
                   Hyb = paste(plotdf$gene_name[plotdf$organism == "hyb"], 
                               plotdf$deletion[plotdf$organism == "hyb"],
                               sign(plotdf$lfc)[plotdf$organism == "hyb"], sep = "_"))) + 
  scale_fill_gradient(low="grey90",high = "red") +
  ggtitle("TF-gene-direction combination is shared")
# TF-gene-direction-timepoint combo is shared:
p_dirTP <- ggVennDiagram(list(Scer = paste(plotdf$gene_name[plotdf$organism == "cer"], 
                                plotdf$deletion[plotdf$organism == "cer"],
                                sign(plotdf$lfc)[plotdf$organism == "cer"],
                                plotdf$timepoint[plotdf$organism == "cer"], sep = "_"), 
                   Spar = paste(plotdf$gene_name[plotdf$organism == "par"], 
                                plotdf$deletion[plotdf$organism == "par"],
                                sign(plotdf$lfc)[plotdf$organism == "par"],
                                plotdf$timepoint[plotdf$organism == "par"], sep = "_"),
                   Hyb = paste(plotdf$gene_name[plotdf$organism == "hyb"], 
                               plotdf$deletion[plotdf$organism == "hyb"],
                               sign(plotdf$lfc)[plotdf$organism == "hyb"],
                                plotdf$timepoint[plotdf$organism == "hyb"], sep = "_"))) + 
  scale_fill_gradient(low="grey90",high = "red") +
  ggtitle("TF-gene-direction-timepoint combination is shared")
```
For the paper figure:
```{r figure_Venntfdel}
# pdf("../Umich_Dissertation/chapters/figures/Networks/VennTFdel.pdf",
#     width = 4, height = 4)
# p
# dev.off()
```

Number of TFs each gene is affected by in each background:
```{r nTF histograms}
plotdf <- expand_grid(gene_name = genedf$gene_name,
                      organism = c("cer", "hyb", "par"))
sigdf <- TFdeldf |> filter(abs(lfc) > eff_thresh & padj < p_thresh)
plotdf$nTFs <- map2(plotdf$gene_name, plotdf$organism, \(g, o) {
  df_go <- sigdf |> filter(gene_name == g & organism == o & allele == o) |> 
    select(deletion) |> unique()
  return(nrow(df_go))
}) |> unlist()
p <- ggplot(plotdf, aes(x = nTFs)) +
  geom_histogram(stat = "count", aes(fill = organism)) +
  scale_fill_discrete(limits = c("cer", "par", "hyb"),
                      type = c(conservation_colordf$value[conservation_colordf$limits == "Scer"],
                               conservation_colordf$value[conservation_colordf$limits == "Spar"],
                               conservation_colordf$value[conservation_colordf$limits == "Hyb"])) +
  facet_wrap(~factor(organism, levels = c("cer", "par", "hyb"),
                     labels = c("Scer", "Spar", "Hyb"))) +
  theme_classic() +
  ylab("Number of genes") +
  xlab("Number of TF knockouts that significantly affect expression of each gene") +
  theme(legend.position = "none")
p
```

For the paper figure:
```{r figure_powerLaw}
# pdf("../Umich_Dissertation/chapters/figures/Networks/nTFs.pdf",
#     width = 9, height = 2)
# p
# dev.off()
```

## "Skyline plots" of TFdel effects per TF


Skyline plots of nUp and nDown per TF (from any timepoint, but each gene is only counted once) for Scer, Spar, and Hyb. Not worrying about whether the effect is shared or not, because the vast majority are not shared:

```{r skylineplots, warning=FALSE}

# Setting a common ylim for comparability
max_nDE <- sigdf |> mutate(lfc_sign = sign(lfc)) |> 
  select(deletion, organism, gene_name, lfc_sign) |> 
  unique() |> group_by(deletion, organism, lfc_sign) |> summarise(n = n()) |>
  select(n) |> pull() |> max()
ylims <- c(-max_nDE+10, max_nDE-10)

# Scer
plotdf_cer <- filter(sigdf, organism == "cer") |> 
  mutate(lfc_sign = sign(lfc)) |> 
  select(gene_name, deletion, lfc_sign) |> 
  unique() # This unique removes double counts of genes that have the same direction of effect from a TFdel at multiple timepoints (still double-counting the cases where a gene is affected at two timepoints in different directions, but this is what we want because we don't want to pick one of those to remove)
p_cer <- plotSkyline(.sigdf = plotdf_cer,
            .title = "S. cer") +
  ylim(ylims)

# Spar
plotdf_par <- filter(sigdf, organism == "par") |> 
  mutate(lfc_sign = sign(lfc)) |> 
  select(gene_name, deletion, lfc_sign) |> 
  unique() 
p_par <- plotSkyline(.sigdf = plotdf_par,
            .title = "S. par") +
  ylim(ylims)

# Hyb
plotdf_hyb <- filter(sigdf, organism == "hyb"& allele == "hyb") |> 
  mutate(lfc_sign = sign(lfc)) |> 
  select(gene_name, deletion, lfc_sign) |> 
  unique() 
p_hyb <- plotSkyline(.sigdf = plotdf_hyb,
            .title = "F1 Hybrid (summed alleles)") +
  ylim(ylims)
```

