<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>DESeq2</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Networks</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">Functions</a>
</li>
<li>
  <a href="data_cleaning.html">Data Cleaning</a>
</li>
<li>
  <a href="DESeq2.html">DESeq2</a>
</li>
<li>
  <a href="Fig_WT.html">Comparing Wildtype LowN Response</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DESeq2</h1>

</div>


<pre class="r"><code>sapply(c(&quot;tidyr&quot;, &quot;dplyr&quot;, &quot;purrr&quot;, &quot;ggplot2&quot;, &quot;ggpubr&quot;, &quot;DESeq2&quot;), require, character.only=TRUE)</code></pre>
<pre><code>##   tidyr   dplyr   purrr ggplot2  ggpubr  DESeq2 
##    TRUE    TRUE    TRUE    TRUE    TRUE    TRUE</code></pre>
<pre class="r"><code>load(&quot;data_files/CleanedCounts.RData&quot;)
eff_thresh &lt;- 1
p_thresh &lt;- 0.05
GeneNames &lt;- rownames(counts)</code></pre>
<pre class="r"><code># data organizing
counts &lt;- list(cer_cer_TP1 = counts[,sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;cer&quot;],
               par_par_TP1 = counts[,sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;par&quot;],
               hyb_cer_TP1 = counts[,sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;cer&quot;],
               hyb_par_TP1 = counts[,sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;par&quot;],
               hyb_hyb_TP1 = counts[,sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;hyb&quot;],
               cer_cer_TP2 = counts[,sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;cer&quot;],
               par_par_TP2 = counts[,sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;par&quot;],
               hyb_cer_TP2 = counts[,sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;cer&quot;],
               hyb_par_TP2 = counts[,sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;par&quot;],
               hyb_hyb_TP2 = counts[,sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;hyb&quot;],
               cer_cer_TP3 = counts[,sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;cer&quot;],
               par_par_TP3 = counts[,sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;par&quot;],
               hyb_cer_TP3 = counts[,sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;cer&quot;],
               hyb_par_TP3 = counts[,sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;par&quot;],
               hyb_hyb_TP3 = counts[,sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;hyb&quot;])

infos &lt;- list(cer_cer_TP1 = sample_info[sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;cer&quot;,],
              par_par_TP1 = sample_info[sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;par&quot;,],
              hyb_cer_TP1 = sample_info[sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;cer&quot;,],
              hyb_par_TP1 = sample_info[sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;par&quot;,],
              hyb_hyb_TP1 = sample_info[sample_info$time_point_num == 0 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;hyb&quot;,],
              cer_cer_TP2 = sample_info[sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;cer&quot;,],
              par_par_TP2 = sample_info[sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;par&quot;,],
              hyb_cer_TP2 = sample_info[sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;cer&quot;,],
              hyb_par_TP2 = sample_info[sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;par&quot;,],
              hyb_hyb_TP2 = sample_info[sample_info$time_point_num == 60 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;hyb&quot;,],
              cer_cer_TP3 = sample_info[sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;cer&quot;,],
              par_par_TP3 = sample_info[sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;par&quot;,],
              hyb_cer_TP3 = sample_info[sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;cer&quot;,],
              hyb_par_TP3 = sample_info[sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;par&quot;,],
              hyb_hyb_TP3 = sample_info[sample_info$time_point_num == 960 &amp; sample_info$organism == &quot;hyb&quot; &amp; sample_info$allele == &quot;hyb&quot;,])</code></pre>
<div id="removing-samples-missing-replicates" class="section level2">
<h2>Removing samples missing replicates</h2>
<p>Rationale: DESeq2 still tries to estimate a log2 fold change when a
genotype has only one replicate. Example: YAP1 in hybrids (any
timepoint). In parents it has 2 replicates:</p>
<pre class="r"><code>infos$cer_cer_TP1 |&gt; select(genotype, time_point_str) |&gt; table()</code></pre>
<pre><code>##               time_point_str
## genotype       0 h, YPD
##   ACE2delete          2
##   AFT1delete          2
##   ARG80delete         2
##   ARG81delete         2
##   BAS1delete          3
##   CBF1delete          4
##   CHA4delete          2
##   DAL80delete         2
##   FKH1delete          2
##   GAT1delete          2
##   GCN4delete          2
##   GCR2delete          2
##   GLN3delete          2
##   GZF3delete          2
##   HAP1delete          2
##   HAP2delete          2
##   HAP3delete          3
##   HAP4delete          2
##   HAP5delete          2
##   INO2delete          2
##   INO4delete          2
##   LEU3delete          2
##   MBP1delete          2
##   MET28delete         5
##   MIG1delete          2
##   MSN2delete          2
##   NRG1delete          2
##   PHD1delete          4
##   PHO4delete          2
##   RFX1delete          2
##   RGT1delete          2
##   RIM101delete        2
##   ROX1delete          2
##   RPN4delete          2
##   RTG1delete          2
##   RTG3delete          2
##   SKN7delete          2
##   SOK2delete          2
##   STB5delete          2
##   SUM1delete          2
##   SWI4delete          3
##   SWI5delete          2
##   TEC1delete          2
##   URE2delete          2
##   WT                 33
##   YAP1delete          2
##   ZAP1delete          1</code></pre>
<p>In hybrids, one:</p>
<pre class="r"><code>infos$hyb_cer_TP1 |&gt; select(genotype, time_point_str) |&gt; table()</code></pre>
<pre><code>##               time_point_str
## genotype       0 h, YPD
##   ACE2delete          2
##   AFT1delete          2
##   ARG80delete         2
##   ARG81delete         2
##   CBF1delete          1
##   CHA4delete          3
##   DAL80delete         2
##   FKH1delete          2
##   GAT1delete          2
##   GCN4delete          1
##   GCR2delete          2
##   GLN3delete          4
##   GZF3delete          2
##   HAP1delete          2
##   HAP2delete          4
##   HAP3delete          2
##   HAP4delete          4
##   HAP5delete          2
##   INO2delete          2
##   INO4delete          2
##   LEU3delete          2
##   MBP1delete          3
##   MET28delete         2
##   MIG1delete          2
##   MSN2delete          2
##   NRG1delete          2
##   PHD1delete          2
##   PHO4delete          2
##   RFX1delete          2
##   RGT1delete          2
##   RIM101delete        2
##   ROX1delete          2
##   RPN4delete          3
##   RTG3delete          1
##   SKN7delete          2
##   SOK2delete          2
##   STB5delete          2
##   SUM1delete          2
##   SWI4delete          2
##   SWI5delete          2
##   TEC1delete          2
##   URE2delete          2
##   WT                 35
##   YAP1delete          1
##   ZAP1delete          3</code></pre>
<p>While most padj are NA, not all are:</p>
<pre class="r"><code># test &lt;- DESeq2::DESeq(DESeqDataSetFromMatrix(countData = counts$hyb_cer_TP1,
#                                      colData = infos$hyb_cer_TP1,
#                                      design = ~ genotype))
# results(test, contrast = c(&quot;genotype&quot;, &quot;YAP1delete&quot;, &quot;WT&quot;),
#         alpha = 0.05)[,&quot;padj&quot;] |&gt; table(useNA = &quot;always&quot;)</code></pre>
<p>Looping through counts/info to remove genotypes that don’t have 2
replicates at a certain timepoint.</p>
<pre class="r"><code># Note: b/c we split the individual timepoints into separate info dataframes, this only removes genotypes from the timepoint where they have no replicates (using a 2d table that includes timepoint is a relic but is helpful for reminding us what timepoint we are looking at)
filterGenotypes &lt;- function(.info, .counts) {
  tab &lt;- .info |&gt; select(genotype, time_point_str) |&gt; table()
  no_rep_genotypes &lt;- rownames(tab)[which(tab &lt; 2)]
  return(filter(.info, !(genotype %in% no_rep_genotypes)))
}
infos &lt;- map(infos, filterGenotypes)
# also updating cols of counts to match columns
updateCountsCols &lt;- function(.counts, .info) {
  return(.counts[,.info$sample_name])
}
counts &lt;- map2(counts, infos, updateCountsCols)

# now YAP1 isn&#39;t in hybrid
infos$cer_cer_TP1 |&gt; select(genotype, time_point_str) |&gt; table()</code></pre>
<pre><code>##               time_point_str
## genotype       0 h, YPD
##   ACE2delete          2
##   AFT1delete          2
##   ARG80delete         2
##   ARG81delete         2
##   BAS1delete          3
##   CBF1delete          4
##   CHA4delete          2
##   DAL80delete         2
##   FKH1delete          2
##   GAT1delete          2
##   GCN4delete          2
##   GCR2delete          2
##   GLN3delete          2
##   GZF3delete          2
##   HAP1delete          2
##   HAP2delete          2
##   HAP3delete          3
##   HAP4delete          2
##   HAP5delete          2
##   INO2delete          2
##   INO4delete          2
##   LEU3delete          2
##   MBP1delete          2
##   MET28delete         5
##   MIG1delete          2
##   MSN2delete          2
##   NRG1delete          2
##   PHD1delete          4
##   PHO4delete          2
##   RFX1delete          2
##   RGT1delete          2
##   RIM101delete        2
##   ROX1delete          2
##   RPN4delete          2
##   RTG1delete          2
##   RTG3delete          2
##   SKN7delete          2
##   SOK2delete          2
##   STB5delete          2
##   SUM1delete          2
##   SWI4delete          3
##   SWI5delete          2
##   TEC1delete          2
##   URE2delete          2
##   WT                 33
##   YAP1delete          2</code></pre>
<pre class="r"><code>infos$hyb_cer_TP1 |&gt; select(genotype, time_point_str) |&gt; table()</code></pre>
<pre><code>##               time_point_str
## genotype       0 h, YPD
##   ACE2delete          2
##   AFT1delete          2
##   ARG80delete         2
##   ARG81delete         2
##   CHA4delete          3
##   DAL80delete         2
##   FKH1delete          2
##   GAT1delete          2
##   GCR2delete          2
##   GLN3delete          4
##   GZF3delete          2
##   HAP1delete          2
##   HAP2delete          4
##   HAP3delete          2
##   HAP4delete          4
##   HAP5delete          2
##   INO2delete          2
##   INO4delete          2
##   LEU3delete          2
##   MBP1delete          3
##   MET28delete         2
##   MIG1delete          2
##   MSN2delete          2
##   NRG1delete          2
##   PHD1delete          2
##   PHO4delete          2
##   RFX1delete          2
##   RGT1delete          2
##   RIM101delete        2
##   ROX1delete          2
##   RPN4delete          3
##   SKN7delete          2
##   SOK2delete          2
##   STB5delete          2
##   SUM1delete          2
##   SWI4delete          2
##   SWI5delete          2
##   TEC1delete          2
##   URE2delete          2
##   WT                 35
##   ZAP1delete          3</code></pre>
<p>Converting genotype to a factor, after removing genotypes, so
different number of levels per dataset.</p>
<pre class="r"><code>infos &lt;- map(infos, mutate, 
             genotype = genotype |&gt; 
               as.factor() |&gt; 
               relevel(ref = &quot;WT&quot;))

# pairing counts and sample info
dds &lt;- map2(counts, infos, \(x, y) {
  output &lt;- DESeqDataSetFromMatrix(countData = x,
                                   colData = y,
                                   design = ~ genotype)
  return(output)
})</code></pre>
<pre><code>## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode</code></pre>
</div>
<div id="model-fitting" class="section level2">
<h2>Model fitting</h2>
<p>This takes the majority of the time:</p>
<pre class="r"><code>library(&quot;BiocParallel&quot;)
register(MulticoreParam(4))
dds &lt;- map(dds, DESeq, parallel = TRUE)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<p>Checking positive control: TDH3 decreases in GCR2delete at TP1 (any
timepoint, any species). We’ll use Scer TP1 as the example.</p>
<pre class="r"><code>test &lt;- results(dds$cer_cer_TP3, contrast = c(&quot;genotype&quot;, &quot;GCR2delete&quot;, &quot;WT&quot;),
                alpha = 0.05)
test[&quot;YGR192C&quot;,] # down in Scer</code></pre>
<pre><code>## log2 fold change (MLE): genotype GCR2delete vs WT 
## Wald test p-value: genotype GCR2delete vs WT 
## DataFrame with 1 row and 6 columns
##          baseMean log2FoldChange     lfcSE      stat
##         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## YGR192C   16876.7       -1.33222  0.142197  -9.36886
##              pvalue        padj
##           &lt;numeric&gt;   &lt;numeric&gt;
## YGR192C 7.33198e-21 7.70151e-18</code></pre>
</div>
<div id="formatting-data" class="section level2">
<h2>Formatting Data</h2>
<pre class="r"><code># extracting LFCs, SEs, and pvalues from each TFdel-WT comparison in each 
# species and output as TFdeldfs
griddfTP1 &lt;- bind_rows(infos$cer_cer_TP1,
                    infos$par_par_TP1,
                    infos$hyb_cer_TP1,
                    infos$hyb_par_TP1,
                    infos$hyb_hyb_TP1) |&gt; 
  select(genotype, allele, organism) |&gt; 
  filter(genotype != &quot;WT&quot;) |&gt; 
  mutate(timepoint = &quot;TP1&quot;) |&gt; 
  unique()
griddfTP2 &lt;- bind_rows(infos$cer_cer_TP2,
                       infos$par_par_TP2,
                       infos$hyb_cer_TP2,
                       infos$hyb_par_TP2,
                       infos$hyb_hyb_TP2) |&gt; 
  select(genotype, allele, organism) |&gt; 
  filter(genotype != &quot;WT&quot;) |&gt; 
  mutate(timepoint = &quot;TP2&quot;) |&gt; 
  unique()
griddfTP3 &lt;- bind_rows(infos$cer_cer_TP3,
                       infos$par_par_TP3,
                       infos$hyb_cer_TP3,
                       infos$hyb_par_TP3,
                       infos$hyb_hyb_TP3) |&gt; 
  select(genotype, allele, organism) |&gt; 
  filter(genotype != &quot;WT&quot;) |&gt; 
  mutate(timepoint = &quot;TP3&quot;) |&gt; 
  unique()
griddf &lt;- bind_rows(griddfTP1, griddfTP2, griddfTP3)

TFdeldf &lt;- map(c(1:nrow(griddf)), \(i) {
  del &lt;- griddf$genotype[i] |&gt; as.character()
  org &lt;- griddf$organism[i]
  al &lt;- griddf$allele[i]
  tp &lt;- griddf$timepoint[i]
  res &lt;- results(dds[[paste(org, al, tp, sep = &quot;_&quot;)]], contrast = c(&quot;genotype&quot;, del, &quot;WT&quot;),
                 alpha = 0.05, cooksCutoff = FALSE, independentFiltering = FALSE)
  return(tibble(gene_name = GeneNames,
                deletion = gsub(&quot;delete&quot;, &quot;&quot;, del),
                organism = org,
                allele = al,
                timepoint = tp,
                basemean = res$baseMean,
                lfc = res$log2FoldChange,
                lfcSE = res$lfcSE,
                pval = res$pvalue,
                padj = res$padj))
}) |&gt; bind_rows()</code></pre>
<p>A few gene/TF/tp combos didn’t converge in DESeq2 or were missing
replicates:</p>
<pre class="r"><code>length(unique(TFdeldf$deletion))</code></pre>
<pre><code>## [1] 46</code></pre>
<pre class="r"><code>sum(is.na(TFdeldf))</code></pre>
<pre><code>## [1] 12812</code></pre>
<pre class="r"><code>sum(is.na(TFdeldf$padj))</code></pre>
<pre><code>## [1] 3203</code></pre>
<pre class="r"><code>table(is.na(TFdeldf$padj), TFdeldf$timepoint)</code></pre>
<pre><code>##        
##             TP1     TP2     TP3
##   FALSE 1108121 1065408 1097617
##   TRUE     1192    1033     978</code></pre>
<pre class="r"><code>sum(is.na(TFdeldf$lfc)) + sum(is.na(TFdeldf$lfcSE)) + sum(is.na(TFdeldf$pval)) + sum(is.na(TFdeldf$padj))</code></pre>
<pre><code>## [1] 12812</code></pre>
<pre class="r"><code>TFdeldf &lt;- drop_na(TFdeldf)
length(unique(TFdeldf$deletion)) # we shouldn&#39;t lose any TFs entirely</code></pre>
<pre><code>## [1] 46</code></pre>
<pre class="r"><code>sum(TFdeldf$padj &lt; p_thresh)</code></pre>
<pre><code>## [1] 126472</code></pre>
<pre class="r"><code>sum(TFdeldf$padj &gt;= p_thresh) # should have way fewer significant effects</code></pre>
<pre><code>## [1] 3144674</code></pre>
</div>
<div
id="negative-control-wt-samples-28-vs-2-leave-one-out-as-sham-tfdels"
class="section level2">
<h2>Negative control: WT samples 28 vs 2, “leave one out” as sham
TFdels</h2>
<p>Rationale: Because there’s such a difference in power between having
2 TFdel replicates versus 30 WT replicates, we want to verify that the 2
TFdel replicates are detecting more and different DE genes than randomly
leaving out 2 WT replicates and treating them like TFdels would.</p>
<pre class="r"><code>set.seed(23)
# looping through counts, change counts for each TFdel to be 2 randomly sampled WT samples (or however many replicates that TF has)
counts_sham1 &lt;- map2(counts, infos, \(x, y) {
  for (del in setdiff(unique(y$genotype), &quot;WT&quot;)) {
    samps &lt;- filter(y, genotype == del) |&gt; select(sample_name) |&gt; pull()
    wt_samps &lt;- filter(y, genotype == &quot;WT&quot;) |&gt; select(sample_name) |&gt; pull()
    x[, samps] &lt;- x[, sample(wt_samps, length(samps), replace = FALSE)] # because we loop through this sampling for every deletion it is effectively sampling with replacement. We just want to make sure the same sample isn&#39;t used twice for the same genotype, as variation for every gene would be 0
  }
  return(x)
})

set.seed(42)

# same thing twice, just randomizing which WT samples go to which TF
counts_sham2 &lt;- map2(counts, infos, \(x, y) {
  for (del in setdiff(unique(y$genotype), &quot;WT&quot;)) {
    samps &lt;- filter(y, genotype == del) |&gt; select(sample_name) |&gt; pull()
    wt_samps &lt;- filter(y, genotype == &quot;WT&quot;) |&gt; select(sample_name) |&gt; pull()
    x[, samps] &lt;- x[, sample(wt_samps, length(samps), replace = FALSE)] # because we loop through this sampling for every deletion it is effectively sampling with replacement. We just want to make sure the same sample isn&#39;t used twice for the same genotype, as variation for every gene would be 0
  }
  return(x)
})

# pairing counts and sample info
dds_sham1 &lt;- map2(counts_sham1, infos, \(x, y) {
  output &lt;- DESeqDataSetFromMatrix(countData = x,
                                   colData = y,
                                   design = ~ genotype)
  return(output)
})</code></pre>
<pre><code>## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode</code></pre>
<pre class="r"><code>dds_sham2 &lt;- map2(counts_sham2, infos, \(x, y) {
  output &lt;- DESeqDataSetFromMatrix(countData = x,
                                   colData = y,
                                   design = ~ genotype)
  return(output)
})</code></pre>
<pre><code>## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode
## converting counts to integer mode</code></pre>
<pre class="r"><code>### Model fitting
library(&quot;BiocParallel&quot;)
register(MulticoreParam(4))
dds_sham1 &lt;- map(dds_sham1, DESeq, parallel = TRUE)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre class="r"><code>dds_sham2 &lt;- map(dds_sham2, DESeq, parallel = TRUE)</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## -- note: fitType=&#39;parametric&#39;, but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType=&#39;local&#39; or &#39;mean&#39; to avoid this message next time.</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## -- replacing outliers and refitting for 1 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates: 4 workers</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates, fitting model and testing: 4 workers</code></pre>
<pre class="r"><code># # checking control: TDH3 decreases in GCR2delete
# # TP1
# test &lt;- results(dds_sham1$cerTP1, contrast = c(&quot;genotype&quot;, &quot;GCR2delete&quot;, &quot;WT&quot;),
#                 alpha = 0.05)
# test[&quot;YGR192C&quot;,] # shouldn&#39;t be DE anymore, WT vs WT

### Formatting data
# extracting LFCs, SEs, and pvalues from each TFdel-WT comparison in each
# species and output as TFdeldfs
TFdeldf_sham1 &lt;- map(c(1:nrow(griddf)), \(i) {
  del &lt;- griddf$genotype[i] |&gt; as.character()
  org &lt;- griddf$organism[i]
  al &lt;- griddf$allele[i]
  tp &lt;- griddf$timepoint[i]
  res &lt;- results(dds_sham1[[paste(org, al, tp, sep = &quot;_&quot;)]], contrast = c(&quot;genotype&quot;, del, &quot;WT&quot;),
                 alpha = 0.05, cooksCutoff = FALSE, independentFiltering = FALSE)
  return(tibble(gene_name = GeneNames,
                deletion = gsub(&quot;delete&quot;, &quot;&quot;, del),
                organism = org,
                allele = al,
                timepoint = tp,
                basemean = res$baseMean,
                lfc = res$log2FoldChange,
                lfcSE = res$lfcSE,
                pval = res$pvalue,
                padj = res$padj))
}) |&gt; bind_rows()

TFdeldf_sham2 &lt;- map(c(1:nrow(griddf)), \(i) {
  del &lt;- griddf$genotype[i] |&gt; as.character()
  org &lt;- griddf$organism[i]
  al &lt;- griddf$allele[i]
  tp &lt;- griddf$timepoint[i]
  res &lt;- results(dds_sham2[[paste(org, al, tp, sep = &quot;_&quot;)]], contrast = c(&quot;genotype&quot;, del, &quot;WT&quot;),
                 alpha = 0.05, cooksCutoff = FALSE, independentFiltering = FALSE)
  return(tibble(gene_name = GeneNames,
                deletion = gsub(&quot;delete&quot;, &quot;&quot;, del),
                organism = org,
                allele = al,
                timepoint = tp,
                basemean = res$baseMean,
                lfc = res$log2FoldChange,
                lfcSE = res$lfcSE,
                pval = res$pvalue,
                padj = res$padj))
}) |&gt; bind_rows()

# A few gene/TF/tp combos didn&#39;t converge in DESeq2 or were missing replicates:
TFdeldf_sham1 &lt;- drop_na(TFdeldf_sham1)
TFdeldf_sham2 &lt;- drop_na(TFdeldf_sham2)

sum(TFdeldf_sham1$padj &lt; p_thresh)</code></pre>
<pre><code>## [1] 782</code></pre>
<pre class="r"><code>sum(TFdeldf_sham1$padj &gt;= p_thresh) # should have way fewer significant effects</code></pre>
<pre><code>## [1] 3265087</code></pre>
<pre class="r"><code>sum(TFdeldf_sham2$padj &lt; p_thresh)</code></pre>
<pre><code>## [1] 1709</code></pre>
<pre class="r"><code>sum(TFdeldf_sham2$padj &gt;= p_thresh)</code></pre>
<pre><code>## [1] 3264160</code></pre>
</div>
<div id="saving" class="section level2">
<h2>Saving</h2>
<pre class="r"><code>save(TFdeldf, file = &quot;data_files/DESeq2.RData&quot;)
# save(TFdeldf, TFdeldf_sham1, TFdeldf_sham2, file = &quot;data_files/DESeq2.RData&quot;)</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
