<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Fig_TFdel</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Networks</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">Functions</a>
</li>
<li>
  <a href="data_cleaning.html">Data Cleaning</a>
</li>
<li>
  <a href="DESeq2.html">DESeq2</a>
</li>
<li>
  <a href="Fig_WT.html">Comparing Wildtype LowN Response</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Fig_TFdel</h1>

</div>


<pre class="r"><code>sapply(c(&quot;dplyr&quot;, &quot;readr&quot;, &quot;tidyr&quot;, &quot;purrr&quot;, &quot;ggplot2&quot;, &quot;ggpubr&quot;, &quot;matrixStats&quot;, &quot;ComplexHeatmap&quot;, &quot;circlize&quot;, &quot;ggVennDiagram&quot;, &quot;ggExtra&quot;), require, character.only=TRUE)</code></pre>
<pre><code>##          dplyr          readr          tidyr          purrr        ggplot2         ggpubr 
##           TRUE           TRUE           TRUE           TRUE           TRUE           TRUE 
##    matrixStats ComplexHeatmap       circlize  ggVennDiagram        ggExtra 
##           TRUE           TRUE           TRUE           TRUE           TRUE</code></pre>
<pre class="r"><code>load(&quot;data_files/CleanedCounts.RData&quot;)
load(&quot;data_files/DESeq2.RData&quot;)
load(&quot;data_files/GeneDataFrame.RData&quot;)
source(&quot;utils.R&quot;)

sgd_lookup &lt;- read_delim(&quot;data_files/downloaded_genomes_and_features/SGD_features.tab&quot;, delim = &quot;\t&quot;, col_names = FALSE) |&gt; filter(X2 == &quot;ORF&quot;) |&gt; 
  select(X4, X5) |&gt; 
  dplyr::rename(&quot;gene_name&quot;=&quot;X4&quot;,
                &quot;common_name&quot;=&quot;X5&quot;)</code></pre>
<pre><code>## Rows: 16454 Columns: 16
## ── Column specification ─────────────────────────────────────────────────────────────────────────────
## Delimiter: &quot;\t&quot;
## chr  (12): X1, X2, X3, X4, X5, X6, X7, X8, X9, X12, X15, X16
## dbl   (3): X10, X11, X13
## date  (1): X14
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>FullTFnames &lt;- setdiff(unique(gsub(&quot;delete&quot;, &quot;&quot;, sample_info$genotype)), &quot;WT&quot;)
NitrogenTFs &lt;- c(&quot;URE2&quot;, &quot;GAT1&quot;, &quot;GLN3&quot;, &quot;DAL80&quot;, &quot;GZF3&quot;)
# ordering TFnames so that nitrogen comes first then alphabetical
TFnames &lt;- c(NitrogenTFs, sort(setdiff(FullTFnames, NitrogenTFs)))

p_thresh &lt;- 0.05 # because DESeq2 already corrected for FDR with alpha = 0.05
eff_thresh &lt;- 2
TFdel_lookup &lt;- read_delim(&quot;data_files/downloaded_genomes_and_features/yeastract_46TFs.csv&quot;, col_names = FALSE, col_select = c(1,2), delim = &quot;;&quot;) # gets some warnings, but so far has been fine</code></pre>
<pre><code>## Rows: 46 Columns: 2
## ── Column specification ─────────────────────────────────────────────────────────────────────────────
## Delimiter: &quot;;&quot;
## chr (2): X1, X2
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>colnames(TFdel_lookup) &lt;- c(&quot;common&quot;, &quot;systematic&quot;)</code></pre>
<div id="which-tfs-are-differentially-expressed-between-species"
class="section level2">
<h2>Which TFs are differentially expressed between species?</h2>
<p>Even though this is only using WT data, this is more relevant to the
TFdels, so we’ll start with this. Mean expression in Scer vs Spar,
averaged across timepoints and replicates. Error bars are standard error
from replicates.</p>
<pre class="r"><code># scatter plot Scer vs Spar with geom_text as the TF names (b/c the heatmap was only useful when there was more than one environment)
plotdf &lt;- tibble(tf = TFnames) |&gt; 
  left_join(y = sgd_lookup, 
            by = c(&quot;tf&quot;=&quot;common_name&quot;))
plotdf$mean_expr_cer &lt;- map(plotdf$gene_name, \(g) {
  expr_vec &lt;- NA
  if (g %in% rownames(counts)) {
    expr_vec &lt;- counts[g, sample_info$genotype == &quot;WT&quot; &amp;
                       sample_info$organism == &quot;cer&quot;] |&gt; 
    as.numeric()
  }
  return(mean(expr_vec))
}) |&gt; unlist()
plotdf$mean_expr_par &lt;- map(plotdf$gene_name, \(g) {
  expr_vec &lt;- NA
  if (g %in% rownames(counts)) {
    expr_vec &lt;- counts[g, sample_info$genotype == &quot;WT&quot; &amp;
                       sample_info$organism == &quot;par&quot;] |&gt; 
    as.numeric()
  }
  return(mean(expr_vec))
}) |&gt; unlist()

print(ggplot(plotdf, aes(x = log2(mean_expr_cer + 1), 
                   y = log2(mean_expr_par + 1))) +
  geom_text(aes(label = tf, color = tf %in% NitrogenTFs)) +
  geom_abline(slope = 1, intercept = 0))</code></pre>
<pre><code>## Warning: Removed 1 row containing missing values or values outside the scale range (`geom_text()`).</code></pre>
<p><img src="Fig_TFdel_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
</div>
<div id="how-many-genes-does-each-tfdel-affect-in-each-species"
class="section level2">
<h2>How many genes does each TFdel affect in each species?</h2>
<p>How many genes are affected in shams vs real knockout lines?</p>
<pre class="r"><code>plotdf &lt;- bind_rows(mutate(TFdeldf, type = &quot;real&quot;),
                    mutate(TFdeldf_sham1, type = &quot;sham1&quot;),
                    mutate(TFdeldf_sham2, type = &quot;sham2&quot;)) |&gt; 
  filter(organism == allele) |&gt; 
  select(gene_name, organism, lfc, padj, type) |&gt; 
  group_by(organism, type) |&gt; 
  summarise(nSig = sum(abs(lfc) &gt; eff_thresh &amp; padj &lt; p_thresh))</code></pre>
<pre><code>## `summarise()` has grouped output by &#39;organism&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p &lt;- ggplot(plotdf, aes(x = type, 
                        y = nSig)) +
  geom_histogram(aes(fill = organism), stat = &quot;identity&quot;) +
  geom_text(aes(x = type, y = nSig, label = nSig), nudge_y = 500) +
  scale_fill_discrete(limits = c(&quot;cer&quot;, &quot;par&quot;, &quot;hyb&quot;),
                      type = c(conservation_colordf$value[conservation_colordf$limits == &quot;Scer&quot;],
                               conservation_colordf$value[conservation_colordf$limits == &quot;Spar&quot;],
                               conservation_colordf$value[conservation_colordf$limits == &quot;Hyb&quot;])) +
  facet_wrap(~factor(organism, levels = c(&quot;cer&quot;, &quot;par&quot;, &quot;hyb&quot;),
                     labels = c(&quot;Scer&quot;, &quot;Spar&quot;, &quot;Hyb&quot;))) +
  theme_classic() +
  xlab(&quot;&quot;) +
  ylab(&quot;Number of significantly differentiall expressed\ngenes across all 46 TF knockouts&quot;) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<pre><code>## Warning in geom_histogram(aes(fill = organism), stat = &quot;identity&quot;): Ignoring unknown parameters:
## `binwidth`, `bins`, and `pad`</code></pre>
<pre class="r"><code>p</code></pre>
<p><img src="Fig_TFdel_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>For the paper figure:</p>
<pre class="r"><code># pdf(&quot;../Umich_Dissertation/chapters/figures/Networks/shams.pdf&quot;,
#     width = 5, height = 4)
# p
# dev.off()</code></pre>
<p>Barplots of nGenes up or down in response to each TF deletion.</p>
<p>If we only consider a TF effect shared if it affects the same gene in
multiple yeasts and occurs at the same timepoint and direction:</p>
<pre class="r"><code>plotdf &lt;- TFdeldf |&gt; filter(padj &lt; p_thresh &amp;
                             abs(lfc) &gt; eff_thresh) |&gt; 
  mutate(lfc_sign = sign(lfc)) |&gt; 
  select(gene_name, deletion, organism, timepoint, lfc_sign) |&gt;
  group_by(gene_name, deletion, timepoint, lfc_sign) |&gt; 
  summarise(orgs = list(organism))</code></pre>
<pre><code>## `summarise()` has grouped output by &#39;gene_name&#39;, &#39;deletion&#39;, &#39;timepoint&#39;. You can override using the
## `.groups` argument.</code></pre>
<pre class="r"><code>cat(&quot;Fraction of single-yeast effects:&quot;, sum(sapply(plotdf$orgs, length) == 1), &quot;/&quot;, nrow(plotdf),
    &quot;(&quot;, sum(sapply(plotdf$orgs, length) == 1)/nrow(plotdf), &quot;%)&quot;, &quot;\n&quot;)</code></pre>
<pre><code>## Fraction of single-yeast effects: 11389 / 15058 ( 0.7563421 %)</code></pre>
<p>If we instead consider a TF effect shared if it affects the same gene
in multiple yeasts in the same direciton, but occurs at any
timepoint:</p>
<pre class="r"><code>plotdf &lt;- TFdeldf |&gt; filter(padj &lt; p_thresh &amp;
                             abs(lfc) &gt; eff_thresh) |&gt; 
  mutate(lfc_sign = sign(lfc)) |&gt; 
  select(gene_name, deletion, organism, lfc_sign) |&gt;
  #unique() |&gt; # unique stops double-counting of gene-deletion-lfc_sign combos that affect multiple timepoints, but that currently results in single-species effects becoming a larger portion of the total effects than the timepoint-specific chunk above, which is counterintuitive so commenting it for now
  group_by(gene_name, deletion, lfc_sign) |&gt; 
  summarise(orgs = list(organism))</code></pre>
<pre><code>## `summarise()` has grouped output by &#39;gene_name&#39;, &#39;deletion&#39;. You can override using the `.groups`
## argument.</code></pre>
<pre class="r"><code>cat(&quot;Fraction of single-yeast effects:&quot;, sum(sapply(plotdf$orgs, length) == 1), &quot;/&quot;, nrow(plotdf),
    &quot;(&quot;, sum(sapply(plotdf$orgs, length) == 1)/nrow(plotdf), &quot;%)&quot;, &quot;\n&quot;)</code></pre>
<pre><code>## Fraction of single-yeast effects: 9720 / 13650 ( 0.7120879 %)</code></pre>
<p>That’s not a very big difference, so for now we’ll specify they’re
shared if they affect the same TF. Venn diagram makes it even clearer
that it doesn’t matter so we might as well choose the loosest, simplest
definition:</p>
<pre class="r"><code>sigdf &lt;- filter(TFdeldf, padj &lt; p_thresh &amp; abs(lfc) &gt; eff_thresh &amp; organism == allele)
plotdf &lt;- sigdf |&gt; filter(organism == allele)
# TF-gene combo is shared:
p &lt;- ggVennDiagram(list(Scer = paste(plotdf$gene_name[plotdf$organism == &quot;cer&quot;], 
                                plotdf$deletion[plotdf$organism == &quot;cer&quot;], sep = &quot;_&quot;), 
                   Spar = paste(plotdf$gene_name[plotdf$organism == &quot;par&quot;], 
                                plotdf$deletion[plotdf$organism == &quot;par&quot;], sep = &quot;_&quot;),
                   Hyb = paste(plotdf$gene_name[plotdf$organism == &quot;hyb&quot;], 
                                plotdf$deletion[plotdf$organism == &quot;hyb&quot;], sep = &quot;_&quot;))) + 
                scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;red&quot;) +
  ggtitle(&quot;TF-gene combination is shared&quot;)
# TF-gene-direction combo is shared:
p_dir &lt;- ggVennDiagram(list(Scer = paste(plotdf$gene_name[plotdf$organism == &quot;cer&quot;], 
                                plotdf$deletion[plotdf$organism == &quot;cer&quot;],
                                sign(plotdf$lfc)[plotdf$organism == &quot;cer&quot;], sep = &quot;_&quot;), 
                   Spar = paste(plotdf$gene_name[plotdf$organism == &quot;par&quot;], 
                                plotdf$deletion[plotdf$organism == &quot;par&quot;],
                                sign(plotdf$lfc)[plotdf$organism == &quot;par&quot;], sep = &quot;_&quot;),
                   Hyb = paste(plotdf$gene_name[plotdf$organism == &quot;hyb&quot;], 
                               plotdf$deletion[plotdf$organism == &quot;hyb&quot;],
                               sign(plotdf$lfc)[plotdf$organism == &quot;hyb&quot;], sep = &quot;_&quot;))) + 
  scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;red&quot;) +
  ggtitle(&quot;TF-gene-direction combination is shared&quot;)
# TF-gene-direction-timepoint combo is shared:
p_dirTP &lt;- ggVennDiagram(list(Scer = paste(plotdf$gene_name[plotdf$organism == &quot;cer&quot;], 
                                plotdf$deletion[plotdf$organism == &quot;cer&quot;],
                                sign(plotdf$lfc)[plotdf$organism == &quot;cer&quot;],
                                plotdf$timepoint[plotdf$organism == &quot;cer&quot;], sep = &quot;_&quot;), 
                   Spar = paste(plotdf$gene_name[plotdf$organism == &quot;par&quot;], 
                                plotdf$deletion[plotdf$organism == &quot;par&quot;],
                                sign(plotdf$lfc)[plotdf$organism == &quot;par&quot;],
                                plotdf$timepoint[plotdf$organism == &quot;par&quot;], sep = &quot;_&quot;),
                   Hyb = paste(plotdf$gene_name[plotdf$organism == &quot;hyb&quot;], 
                               plotdf$deletion[plotdf$organism == &quot;hyb&quot;],
                               sign(plotdf$lfc)[plotdf$organism == &quot;hyb&quot;],
                                plotdf$timepoint[plotdf$organism == &quot;hyb&quot;], sep = &quot;_&quot;))) + 
  scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;red&quot;) +
  ggtitle(&quot;TF-gene-direction-timepoint combination is shared&quot;)</code></pre>
<p>For the paper figure:</p>
<pre class="r"><code># pdf(&quot;../Umich_Dissertation/chapters/figures/Networks/VennTFdel.pdf&quot;,
#     width = 4, height = 4)
# p
# dev.off()</code></pre>
<p>Number of TFs each gene is affected by in each background:</p>
<pre class="r"><code>plotdf &lt;- expand_grid(gene_name = genedf$gene_name,
                      organism = c(&quot;cer&quot;, &quot;hyb&quot;, &quot;par&quot;))
sigdf &lt;- TFdeldf |&gt; filter(abs(lfc) &gt; eff_thresh &amp; padj &lt; p_thresh)
plotdf$nTFs &lt;- map2(plotdf$gene_name, plotdf$organism, \(g, o) {
  df_go &lt;- sigdf |&gt; filter(gene_name == g &amp; organism == o &amp; allele == o) |&gt; 
    select(deletion) |&gt; unique()
  return(nrow(df_go))
}) |&gt; unlist()
p &lt;- ggplot(plotdf, aes(x = nTFs)) +
  geom_histogram(stat = &quot;count&quot;, aes(fill = organism)) +
  scale_fill_discrete(limits = c(&quot;cer&quot;, &quot;par&quot;, &quot;hyb&quot;),
                      type = c(conservation_colordf$value[conservation_colordf$limits == &quot;Scer&quot;],
                               conservation_colordf$value[conservation_colordf$limits == &quot;Spar&quot;],
                               conservation_colordf$value[conservation_colordf$limits == &quot;Hyb&quot;])) +
  facet_wrap(~factor(organism, levels = c(&quot;cer&quot;, &quot;par&quot;, &quot;hyb&quot;),
                     labels = c(&quot;Scer&quot;, &quot;Spar&quot;, &quot;Hyb&quot;))) +
  theme_classic() +
  ylab(&quot;Number of genes&quot;) +
  xlab(&quot;Number of TF knockouts that significantly affect expression of each gene&quot;) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<pre><code>## Warning in geom_histogram(stat = &quot;count&quot;, aes(fill = organism)): Ignoring unknown parameters:
## `binwidth`, `bins`, and `pad`</code></pre>
<pre class="r"><code>p</code></pre>
<p><img src="Fig_TFdel_files/figure-html/nTF%20histograms-1.png" width="672" /></p>
<p>For the paper figure:</p>
<pre class="r"><code># pdf(&quot;../Umich_Dissertation/chapters/figures/Networks/nTFs.pdf&quot;,
#     width = 9, height = 2)
# p
# dev.off()</code></pre>
</div>
<div id="skyline-plots-of-tfdel-effects-per-tf" class="section level2">
<h2>“Skyline plots” of TFdel effects per TF</h2>
<p>Skyline plots of nUp and nDown per TF (from any timepoint, but each
gene is only counted once) for Scer, Spar, and Hyb. Not worrying about
whether the effect is shared or not, because the vast majority are not
shared:</p>
<pre class="r"><code># Setting a common ylim for comparability
max_nDE &lt;- sigdf |&gt; mutate(lfc_sign = sign(lfc)) |&gt; 
  select(deletion, organism, gene_name, lfc_sign) |&gt; 
  unique() |&gt; group_by(deletion, organism, lfc_sign) |&gt; summarise(n = n()) |&gt;
  select(n) |&gt; pull() |&gt; max()</code></pre>
<pre><code>## `summarise()` has grouped output by &#39;deletion&#39;, &#39;organism&#39;. You can override using the `.groups`
## argument.
## Adding missing grouping variables: `deletion`, `organism`</code></pre>
<pre class="r"><code>ylims &lt;- c(-max_nDE+10, max_nDE-10)

# Scer
plotdf_cer &lt;- filter(sigdf, organism == &quot;cer&quot;) |&gt; 
  mutate(lfc_sign = sign(lfc)) |&gt; 
  select(gene_name, deletion, lfc_sign) |&gt; 
  unique() # This unique removes double counts of genes that have the same direction of effect from a TFdel at multiple timepoints (still double-counting the cases where a gene is affected at two timepoints in different directions, but this is what we want because we don&#39;t want to pick one of those to remove)
p_cer &lt;- plotSkyline(.sigdf = plotdf_cer,
            .title = &quot;S. cer&quot;) +
  ylim(ylims)

# Spar
plotdf_par &lt;- filter(sigdf, organism == &quot;par&quot;) |&gt; 
  mutate(lfc_sign = sign(lfc)) |&gt; 
  select(gene_name, deletion, lfc_sign) |&gt; 
  unique() 
p_par &lt;- plotSkyline(.sigdf = plotdf_par,
            .title = &quot;S. par&quot;) +
  ylim(ylims)

# Hyb
plotdf_hyb &lt;- filter(sigdf, organism == &quot;hyb&quot;&amp; allele == &quot;hyb&quot;) |&gt; 
  mutate(lfc_sign = sign(lfc)) |&gt; 
  select(gene_name, deletion, lfc_sign) |&gt; 
  unique() 
p_hyb &lt;- plotSkyline(.sigdf = plotdf_hyb,
            .title = &quot;F1 Hybrid (summed alleles)&quot;) +
  ylim(ylims)</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
