<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Wildtype differences in Low Nitrogen Network between Scer and Spar</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Networks</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">Functions</a>
</li>
<li>
  <a href="data_cleaning.html">Data Cleaning</a>
</li>
<li>
  <a href="DESeq2.html">DESeq2</a>
</li>
<li>
  <a href="Fig_WT.html">Comparing Wildtype LowN Response</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Wildtype differences in Low Nitrogen
Network between Scer and Spar</h1>

</div>


<pre class="r"><code>sapply(c(&quot;dplyr&quot;, &quot;readr&quot;, &quot;tidyr&quot;, &quot;purrr&quot;, &quot;ggplot2&quot;, &quot;ggpubr&quot;, &quot;matrixStats&quot;, &quot;ggVennDiagram&quot;), require, character.only=TRUE)</code></pre>
<pre><code>##         dplyr         readr         tidyr         purrr       ggplot2 
##          TRUE          TRUE          TRUE          TRUE          TRUE 
##        ggpubr   matrixStats ggVennDiagram 
##          TRUE          TRUE          TRUE</code></pre>
<pre class="r"><code>load(&quot;data_files/CleanedCounts.RData&quot;)

TFnames &lt;- setdiff(unique(gsub(&quot;delete&quot;, &quot;&quot;, sample_info$genotype)), &quot;WT&quot;)
p_thresh &lt;- 0.05 # because DESeq2 already corrected for FDR with alpha = 0.05
eff_thresh &lt;- 1
TFdel_lookup &lt;- read_delim(&quot;data_files/downloaded_genomes_and_features/yeastract_46TFs.csv&quot;, col_names = FALSE, col_select = c(1,2), delim = &quot;;&quot;)
colnames(TFdel_lookup) &lt;- c(&quot;common&quot;, &quot;systematic&quot;)</code></pre>
<div id="classifying-genes-by-their-nitrogen-response"
class="section level2">
<h2>Classifying genes by their nitrogen response</h2>
<pre class="r"><code># First helper function for classifyNitrogenResponse
# Given replicate expression at one timepoint,
# returns mean and 95% confidence interval
measureCI &lt;- function(.rep_expr) {
  mean_expr &lt;- mean(.rep_expr)
  sd_expr &lt;- sd(.rep_expr)
  return(list(mean = mean_expr,
              upper = mean_expr + 1.96*sd_expr,
              lower = mean_expr - 1.96*sd_expr))
}
# # tests for measureCI
# measureCI(1) |&gt; unlist()
# measureCI(rnorm(10, mean = 10, sd = 7)) |&gt; unlist()</code></pre>
<pre class="r"><code># Second helper function
# Given pairs of confidence intervals calculated by measureCI, returns TRUE if CIs overlap or FALSE if CI_ref is greater or less than CI_test) 
areOverlapingCIs &lt;- function(.CI_ref, .CI_test) {
  # check ref &gt; test
  if (.CI_ref$lower &gt; .CI_test$upper) {
    return(FALSE)
  }
  # check ref &lt; test
  if (.CI_ref$upper &lt; .CI_test$lower) {
    return(FALSE)
  }
  else {
    return(TRUE)
  }
}
# # tests for areOverlapingCIs
# test_ref_ci &lt;- measureCI(rnorm(10, mean = 10, sd = 7))
# test_test_ci &lt;- measureCI(rnorm(10, mean = 20, sd = 7))
# areOverlapingCIs(test_ref_ci, test_test_ci) # should be overlapping
# test_ref_ci &lt;- measureCI(rnorm(10, mean = 10, sd = 3))
# test_test_ci &lt;- measureCI(rnorm(10, mean = 20, sd = 2))
# areOverlapingCIs(test_ref_ci, test_test_ci) # shouldn&#39;t be overlapping
# test_ref_ci &lt;- measureCI(1)
# test_test_ci &lt;- measureCI(2)
# areOverlapingCIs(test_ref_ci, test_test_ci) # needs replicate measurements</code></pre>
<pre class="r"><code>classifyNitrogenResponse &lt;- function(.expr_vec, .timepoint_vec) {
  timepoints &lt;- unique(.timepoint_vec) |&gt; sort(decreasing = FALSE)
  if (length(timepoints) != 3) {
    stop(&quot;All three timepoints are not represented.\n Represented timepoints:&quot;,
         timepoints, &quot;\n&quot;)
  }
  tp1CI &lt;- measureCI(.expr_vec[.timepoint_vec == timepoints[1]])
  tp2CI &lt;- measureCI(.expr_vec[.timepoint_vec == timepoints[2]])
  tp3CI &lt;- measureCI(.expr_vec[.timepoint_vec == timepoints[3]])
  # check TP1 vs TP2
  result12 &lt;- areOverlapingCIs(tp1CI, tp2CI)
  # check TP2 vs TP3
  result23 &lt;- areOverlapingCIs(tp2CI, tp3CI)
  # check TP1 vs TP3
  result13 &lt;- areOverlapingCIs(tp1CI, tp3CI)
  if (result12 &amp; result23) {
    if (result13) {
      return(&quot;none&quot;)
    }
    if (!result13) {
      return(&quot;gradual&quot;)
    }
  }
  if (!result12) {
    if (result23) {
      return(&quot;early&quot;)
    }
    if (!result23) {
      return(&quot;both&quot;)
    }
  }
  if (!result23) {
    if (result12) {
      return(&quot;late&quot;)
    }
    if (!result12) {
      return(&quot;fitzwilliam&quot;) # (both, but we shouldn&#39;t ever reach this line)
    }
  }
}
# tests for classifyNitrogenResponse
test_org &lt;- sample(c(&quot;cer&quot;, &quot;par&quot;, &quot;hyb&quot;), 1)
test_gene &lt;- sample(rownames(counts), 1)
test_expr &lt;- counts[test_gene, sample_info$organism == test_org &amp; sample_info$genotype == &quot;WT&quot;] |&gt; as.numeric()
test_timepoints &lt;- sample_info[sample_info$organism == test_org &amp; sample_info$genotype == &quot;WT&quot;, &quot;time_point_num&quot;] |&gt; as.numeric()
classifyNitrogenResponse(.expr_vec = test_expr,
                         .timepoint_vec = test_timepoints)</code></pre>
<pre><code>## [1] &quot;gradual&quot;</code></pre>
<p>Plot to see if we visually agree with the test result:</p>
<pre class="r"><code>plotGenesTFdel(.gene_idxs = test_gene, .tf = &quot;GCR2&quot;, .show_tfdel = FALSE,
               .normalization = &quot;none&quot;, .parents_or_hybrid = &quot;parents&quot;)</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code>plotGenesTFdel(.gene_idxs = test_gene, .tf = &quot;GCR2&quot;, .show_tfdel = FALSE,
               .normalization = &quot;none&quot;, .parents_or_hybrid = &quot;hybrid&quot;)</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-2-2.png" width="672" />
Applying to all genes:</p>
<pre class="r"><code>responsedf &lt;- expand_grid(organism = unique(sample_info$organism),
                          gene_name = rownames(counts))
responsedf$response &lt;- map(c(1:nrow(responsedf)), \(i) {
  org &lt;- responsedf$organism[i]
  g &lt;- responsedf$gene_name[i]
  expr_vec &lt;- counts[g, sample_info$organism == org &amp;
                       sample_info$allele == org &amp;
                       sample_info$genotype == &quot;WT&quot;] |&gt; as.numeric()
  time_vec &lt;- sample_info[sample_info$organism == org &amp;
                            sample_info$allele == org &amp;
                            sample_info$genotype == &quot;WT&quot;, &quot;time_point_num&quot;] |&gt; as.numeric()
  classifyNitrogenResponse(.expr_vec = expr_vec,
                           .timepoint_vec = time_vec)
}) |&gt; unlist()</code></pre>
<p>How many genes in each response group?</p>
<pre class="r"><code>ggplot(responsedf, aes(x = factor(response,
                                  levels = c(&quot;none&quot;, &quot;gradual&quot;, &quot;early&quot;, &quot;late&quot;, &quot;both&quot;)))) +
  geom_histogram(aes(fill = response), stat = &quot;count&quot;) +
  stat_count(binwidth = 1, 
             geom = &#39;text&#39;, 
             color = &#39;black&#39;, 
             aes(label = after_stat(count)),
           position = position_stack(vjust = 1)) +
  facet_wrap(~factor(organism,
                     levels = c(&quot;cer&quot;, &quot;par&quot;, &quot;hyb&quot;),
                     labels = c(&quot;Scer&quot;, &quot;Spar&quot;, &quot;Hyb&quot;))) + 
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90)) +
  xlab(&quot;response&quot;)</code></pre>
<pre><code>## Warning in geom_histogram(aes(fill = response), stat = &quot;count&quot;): Ignoring
## unknown parameters: `binwidth`, `bins`, and `pad`</code></pre>
<pre><code>## Warning in stat_count(binwidth = 1, geom = &quot;text&quot;, color = &quot;black&quot;,
## aes(label = after_stat(count)), : Ignoring unknown parameters: `binwidth`</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-4-1.png" width="672" />
What percent of genes are conserved unresponsive in all 3 groups?</p>
<pre class="r"><code>genedf &lt;- responsedf |&gt; 
  pivot_wider(id_cols = &quot;gene_name&quot;, 
              names_from = &quot;organism&quot;, 
              values_from = &quot;response&quot;) |&gt; 
  mutate(is_conserved_unresponsive = (cer == &quot;none&quot;) &amp;
           (par == &quot;none&quot;) &amp; (hyb == &quot;none&quot;))
sum(genedf$is_conserved_unresponsive)/nrow(genedf)</code></pre>
<pre><code>## [1] 0.7689868</code></pre>
<p>Which (responsive) genes are the same response in Scer and Spar?</p>
<pre class="r"><code>genedf |&gt; filter(!is_conserved_unresponsive) |&gt; 
  select(cer, par) |&gt; table()</code></pre>
<pre><code>##          par
## cer       both early gradual late none
##   both       0     0       2    0    0
##   early      4    23      90    4   79
##   gradual    3    15     195   39  317
##   late       2     0       6   21   22
##   none       2    21     138   25  230</code></pre>
<pre class="r"><code>genedf |&gt; filter(!is_conserved_unresponsive) |&gt; 
  select(cer, hyb) |&gt; table()</code></pre>
<pre><code>##          hyb
## cer       both early gradual late none
##   both       0     1       1    0    0
##   early      0   137      38    1   24
##   gradual    1    77     357   23  111
##   late       0     1       6   40    4
##   none       4    52     291   11   58</code></pre>
<pre class="r"><code>genedf |&gt; filter(!is_conserved_unresponsive) |&gt; 
  select(par, hyb) |&gt; table()</code></pre>
<pre><code>##          hyb
## par       both early gradual late none
##   both       1     8       0    2    0
##   early      0    43       9    0    7
##   gradual    0   127     244    9   51
##   late       3     3      33   42    8
##   none       1    87     407   22  131</code></pre>
<p>Numbers for 3-ring venn diagram of conserved vs diverged responsive
genes</p>
<pre class="r"><code>plotdf &lt;- responsedf |&gt; filter(response != &quot;none&quot;)
ggVennDiagram(list(Scer = plotdf$gene_name[plotdf$organism == &quot;cer&quot;], 
                   Spar = plotdf$gene_name[plotdf$organism == &quot;par&quot;],
                   Hyb = plotdf$gene_name[plotdf$organism == &quot;hyb&quot;])) + 
                scale_fill_gradient(low=&quot;grey90&quot;,high = &quot;red&quot;)</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
<div id="ortholog-correlations" class="section level2">
<h2>Ortholog correlations</h2>
<p>Correlation is calculated between the three timepoints, average
expression between replicates (because pairing replicates only works for
hybrid alleles that we know are from the same cell)</p>
<pre class="r"><code>collapsed &lt;- expand_grid(gene_name = rownames(counts),
                         time_point_num = unique(sample_info$time_point_num),
                         organism = unique(sample_info$organism),
                         allele = unique(sample_info$allele))
collapsed$mean_expr &lt;- map(c(1:nrow(collapsed)), \(i) {
  g &lt;- collapsed$gene_name[i]
  org &lt;- collapsed$organism[i]
  al &lt;- collapsed$allele[i]
  tp &lt;- collapsed$time_point_num[i]
  expr_vec &lt;- counts[g, sample_info$genotype == &quot;WT&quot; &amp;
                       sample_info$organism == org &amp;
                       sample_info$allele == al &amp;
                       sample_info$time_point_num == tp]
  return(mean(expr_vec))
}) |&gt; unlist()
collapsed &lt;- drop_na(collapsed) # NAs are combinations of org/allele that don&#39;t exist (i.e. spar allele in scer org)</code></pre>
<p>Giving each gene its correlation, either between parents (cor(Scer,
Spar)), hybrid alleles (cor(Hyc, Hyp)), Scer-hyb ((cor(Scer, Hyb))), or
Spar-hyb (cor(Spar, Hyb)).</p>
<pre class="r"><code>getOrthoCor &lt;- function(.g, .org1, .al1, .org2, .al2) {
  expr1 &lt;- collapsed |&gt; filter(gene_name == .g &amp;
                                 organism == .org1 &amp;
                                 allele == .al1) |&gt; 
    dplyr::rename(&quot;mean_expr1&quot;=&quot;mean_expr&quot;) |&gt; 
    select(gene_name, time_point_num, mean_expr1)
  expr2 &lt;- collapsed |&gt; filter(gene_name == .g &amp;
                                 organism == .org2 &amp;
                                 allele == .al2) |&gt; 
    dplyr::rename(&quot;mean_expr2&quot;=&quot;mean_expr&quot;) |&gt; 
    select(gene_name, time_point_num, mean_expr2)
  outdf &lt;- inner_join(expr1, expr2,
                      by = c(&quot;gene_name&quot;, &quot;time_point_num&quot;))
  return(cor(outdf$mean_expr1, outdf$mean_expr2))
}
# tests for getOrthoCor
getOrthoCor(.g = &quot;YGR192C&quot;,
            .org1 = &quot;cer&quot;, .al1 = &quot;cer&quot;,
            .org2 = &quot;par&quot;, .al2 = &quot;par&quot;)</code></pre>
<pre><code>## [1] 0.2430612</code></pre>
<p>Plot to see if we visually agree with the test result:</p>
<pre class="r"><code>plotGenesTFdel(.gene_idxs = &quot;YGR192C&quot;, .tf = &quot;GCR2&quot;, .show_tfdel = FALSE,
               .normalization = &quot;none&quot;, .parents_or_hybrid = &quot;parents&quot;)</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-10-1.png" width="672" />
Calculating ortholog correlations:</p>
<pre class="r"><code># cor(Scer, Spar)
genedf$cor_parents &lt;- map(genedf$gene_name, 
                          .f = getOrthoCor, 
                          .org1 = &quot;cer&quot;, .al1 = &quot;cer&quot;, 
                          .org2 = &quot;par&quot;, .al2 = &quot;par&quot;) |&gt; 
  unlist()
# cor(Hyc, Hyp)
genedf$cor_hybrid &lt;- map(genedf$gene_name, 
                         .f = getOrthoCor, 
                         .org1 = &quot;hyb&quot;, .al1 = &quot;cer&quot;, 
                         .org2 = &quot;hyb&quot;, .al2 = &quot;par&quot;) |&gt; 
  unlist()
# cor(Scer, Hyb)
genedf$cor_cer &lt;- map(genedf$gene_name, 
                      .f = getOrthoCor, 
                      .org1 = &quot;cer&quot;, .al1 = &quot;cer&quot;, 
                      .org2 = &quot;hyb&quot;, .al2 = &quot;hyb&quot;) |&gt; 
  unlist()
# cor(Spar, Hyb)
genedf$cor_par &lt;- map(genedf$gene_name, 
                      .f = getOrthoCor, 
                      .org1 = &quot;par&quot;, .al1 = &quot;par&quot;, 
                      .org2 = &quot;hyb&quot;, .al2 = &quot;hyb&quot;) |&gt; 
  unlist()</code></pre>
<p>Genes with low correlation in the parents tend to have high
correlation in the hybrid</p>
<pre class="r"><code>cor_thresh &lt;- 0.5
# parents
plotdf &lt;- genedf |&gt; drop_na() |&gt; 
  filter(!is_conserved_unresponsive)
ggplot(plotdf, aes(x = cor_parents)) +
  geom_density() +
  geom_vline(xintercept = cor_thresh) +
  xlab(&quot;parental correlation between\n orthologs (cor(Scer, Spar))&quot;) +
  theme_classic()</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code>plotdf &lt;- genedf |&gt; drop_na() |&gt; 
  filter(cor_parents &lt; cor_thresh &amp;
           !is_conserved_unresponsive) |&gt; 
  pivot_longer(cols = c(&quot;cor_hybrid&quot;,
                        &quot;cor_parents&quot;),
               names_to = &quot;parent_or_hybrid&quot;,
               values_to = &quot;cor&quot;)
ggplot(plotdf, aes(x = cor)) +
  geom_density(aes(fill = factor(parent_or_hybrid,
                                 levels = c(&quot;cor_parents&quot;,
                                            &quot;cor_hybrid&quot;),
                                 labels = c(&quot;parent&quot;, 
                                            &quot;hybrid&quot;))),
               alpha = 0.5) +
  xlab(&quot;correlation for genes \nwith low parental correlation&quot;) +
  theme_classic() +
  theme(legend.title = element_blank())</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-12-2.png" width="672" />
Furthermore these genes tend to correlate with one or the other parent’s
expression:</p>
<pre class="r"><code>sgd_lookup &lt;- read_delim(&quot;data_files/downloaded_genomes_and_features/SGD_features.tab&quot;, delim = &quot;\t&quot;, col_names = FALSE) |&gt; filter(X2 == &quot;ORF&quot;) |&gt; 
  select(X4, X5) |&gt; 
  dplyr::rename(&quot;gene_name&quot;=&quot;X4&quot;,
                &quot;common_name&quot;=&quot;X5&quot;)</code></pre>
<pre><code>## Rows: 16454 Columns: 16
## ── Column specification ──────────────────────────────────────────────────
## Delimiter: &quot;\t&quot;
## chr  (12): X1, X2, X3, X4, X5, X6, X7, X8, X9, X12, X15, X16
## dbl   (3): X10, X11, X13
## date  (1): X14
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>plotdf &lt;- genedf |&gt; drop_na() |&gt; 
  filter(cor_parents &lt; cor_thresh &amp;
           !is_conserved_unresponsive) |&gt; 
  mutate(conserved_btwn_species = cer == par) |&gt; 
  left_join(y = sgd_lookup, by = &quot;gene_name&quot;)
ggplot(plotdf, aes(x = cor_cer, y = cor_par)) +
  geom_point() +
  geom_vline(xintercept = 0.8, color = &quot;red&quot;) +
  geom_hline(yintercept = 0.8, color = &quot;red&quot;) +
  geom_text(data = filter(plotdf, cor_cer &lt; 0.7 &amp;
                            cor_par &lt; 0.7),
            aes(label = common_name, x = cor_cer,
                y = cor_par),
            nudge_y = -0.1) +
  theme_classic() +
  xlab(&quot;correlation (Scer, Hyb)&quot;) +
  ylab(&quot;correlation (Spar, Hyb)&quot;)</code></pre>
<p><img src="Fig_WT_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<pre class="r"><code>save(genedf, file = &quot;data_files/GeneDataFrame.RData&quot;)</code></pre>
<pre class="r"><code># TODO: adapt to use TFdeldf if needed
# 
# TFdf &lt;- TFdeldf |&gt; select(gene_name, deletion, timepoint, organism, lfc, padj) |&gt;
#   pivot_wider(id_cols = c(&quot;gene_name&quot;, &quot;deletion&quot;, &quot;timepoint&quot;),
#               names_from = &quot;organism&quot;, values_from = c(&quot;lfc&quot;, &quot;padj&quot;)) |&gt; 
#   right_join(y = TFdel_lookup, by = c(&quot;gene_name&quot;=&quot;systematic&quot;),
#              relationship = &quot;many-to-one&quot;)

# # heatmap of TF lfc between species in each environment
# effectsdf &lt;- TFdf |&gt; 
#   drop_na() |&gt; 
#   select(common, experiment, effect_size_species) |&gt; 
#   pivot_wider(id_cols = &quot;common&quot;, names_from = &quot;experiment&quot;,
#               values_from = &quot;effect_size_species&quot;)
# effects_mat &lt;- select(effectsdf, -common) |&gt; as.matrix()
# rownames(effects_mat) &lt;- effectsdf$common
# effects_mat[is.na(effects_mat)] &lt;- 0 # Heatmap can&#39;t handle the number of NAs
# pvalsdf &lt;- TFdf |&gt; 
#   drop_na() |&gt; 
#   select(common, experiment, pvalue_species) |&gt; 
#   pivot_wider(id_cols = &quot;common&quot;, names_from = &quot;experiment&quot;,
#               values_from = &quot;pvalue_species&quot;)
# pvals_mat &lt;- select(pvalsdf, -common) |&gt; as.matrix()
# rownames(pvals_mat) &lt;- pvalsdf$common
# pvals_mat[is.na(pvals_mat)] &lt;- 1
# 
# col_fun &lt;- colorRamp2(c(-2, 0, 2), c(&quot;blue2&quot;, &quot;lightyellow&quot;, &quot;orange1&quot;))
# Heatmap(effects_mat, col = col_fun, na_col = &quot;grey&quot;,
#         column_order = c(&quot;HAP4&quot;, &quot;CC&quot;, &quot;LowN&quot;, &quot;LowPi&quot;, &quot;Heat&quot;, &quot;Cold&quot;),
#         cell_fun = function(j, i, x, y, width, height, fill) {
#   output &lt;- ifelse(pvals_mat[i, j] &lt; 1e-5, yes = &quot;*&quot;, no = &quot;&quot;)
#     grid.text(output, x, y, gp = gpar(fontsize = 10))
# })
# # GAT1, PHD1, INO4, MBP1, AFT1 most obviously higher expressed in Scer, 
# # PHO4, TEC1, MET28 higher in Spar
# # mostly consistent btwn experiments with some exceptions mainly in heat/cold
# # HAP4, RPN4 most dramatic difference in effect direction---Scer up in Heat, Spar up in all others</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
