<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Quality Control</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Networks</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">Functions</a>
</li>
<li>
  <a href="data_cleaning.html">Data Cleaning</a>
</li>
<li>
  <a href="DESeq2.html">DESeq2</a>
</li>
<li>
  <a href="Fig_WT.html">Comparing WT LowN Response</a>
</li>
<li>
  <a href="Fig_TFdel.html">Effects of TF deletions</a>
</li>
<li>
  <a href="Fig_Alleleic.html">Hybrid alleleic expression</a>
</li>
<li>
  <a href="Fig_Variance.html">TF deletions reduce expression variation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Quality Control</h1>

</div>


<div
id="qc-comparing-power-to-detect-de-genes-upon-tf-deletion-in-each-species"
class="section level1">
<h1>QC: comparing power to detect DE genes upon TF deletion in each
species</h1>
<p>TODO: use anything useful in this script for QC of power in different
organisms, but donâ€™t get bogged down in all the complexities of
comparing models etc.</p>
<pre class="r"><code># sapply(c(&quot;dplyr&quot;, &quot;purrr&quot;, &quot;tidyr&quot;, &quot;ggpubr&quot;, &quot;readr&quot;, &quot;data.table&quot;, &quot;ggplot2&quot;, &quot;data.table&quot;, &quot;msir&quot;, &quot;WGCNA&quot;, &quot;energy&quot;), require, character.only=TRUE)
# setwd(&quot;/Users/annar/Documents/Wittkopp_Lab/networks/DDivergence/Redhuis2025/&quot;)
# library(MASS, include.only = &quot;glm.nb&quot;)
# 
# # set constants
# coef_thresh &lt;- 1 # minimum magnitude, so if it&#39;s &lt; -coef_thresh, we also want that to be detected
# p_thresh &lt;- 1e-5
# 
# # Goal #1: recapitulate power issue in non-DESeq2 models, examples:
# # YAP1 TP3 has stronger power in Scer than Spar
# # MET28 TP1 has stronger power in Spar than Scer
# 
# ### Cleaning data
# load(file = &quot;data_files/TFdel.RData&quot;)
# 
# # verify that counts are counts per million
# # Should have very little variation around libsize of 1mil
# hist(colSums(counts_tfdel, na.rm = TRUE))
# hist(colSums(counts_tfdel_allele, na.rm = TRUE))
# 
# infos_list &lt;- list(&quot;cer_TP1&quot; = filter(sample_info_tfdel, allele == &quot;cer&quot; &amp;
#                                         time_point_str == &quot;0 h, YPD&quot;),
#                    &quot;cer_TP2&quot; = filter(sample_info_tfdel, allele == &quot;cer&quot; &amp;
#                                         time_point_str == &quot;1 h, low N&quot;),
#                    &quot;cer_TP3&quot; = filter(sample_info_tfdel, allele == &quot;cer&quot; &amp;
#                                         time_point_str == &quot;16 h, low N&quot;),
#                    &quot;par_TP1&quot; = filter(sample_info_tfdel, allele == &quot;par&quot; &amp;
#                                         time_point_str == &quot;0 h, YPD&quot;),
#                    &quot;par_TP2&quot; = filter(sample_info_tfdel, allele == &quot;par&quot; &amp;
#                                         time_point_str == &quot;1 h, low N&quot;),
#                    &quot;par_TP3&quot; = filter(sample_info_tfdel, allele == &quot;par&quot; &amp;
#                                         time_point_str == &quot;16 h, low N&quot;),
#                    &quot;hyc_TP1&quot; = filter(sample_info_tfdel_allele, allele == &quot;cer&quot; &amp;
#                                         time_point_str == &quot;0 h, YPD&quot;),
#                    &quot;hyc_TP2&quot; = filter(sample_info_tfdel_allele, allele == &quot;cer&quot; &amp;
#                                         time_point_str == &quot;1 h, low N&quot;),
#                    &quot;hyc_TP3&quot; = filter(sample_info_tfdel_allele, allele == &quot;cer&quot; &amp;
#                                         time_point_str == &quot;16 h, low N&quot;),
#                    &quot;hyp_TP1&quot; = filter(sample_info_tfdel_allele, allele == &quot;par&quot; &amp;
#                                         time_point_str == &quot;0 h, YPD&quot;),
#                    &quot;hyp_TP2&quot; = filter(sample_info_tfdel_allele, allele == &quot;par&quot; &amp;
#                                         time_point_str == &quot;1 h, low N&quot;),
#                    &quot;hyp_TP3&quot; = filter(sample_info_tfdel_allele, allele == &quot;par&quot; &amp;
#                                         time_point_str == &quot;16 h, low N&quot;))
# 
# counts_list &lt;- map(infos_list, \(x) {
#   samples &lt;- select(x, sample_name) |&gt; pull()
#   cts &lt;- cbind(counts_tfdel, counts_tfdel_allele)
#   return(cts[, samples])
# })
# 
# # test data : YAP1 in TP3, MET28 in TP1
# infos_list_test &lt;- list(&quot;cer_YAP1&quot; = filter(infos_list$cer_TP3,
#                                             genotype %in% c(&quot;WT&quot;, &quot;YAP1delete&quot;)),
#                         &quot;par_YAP1&quot; = filter(infos_list$par_TP3,
#                                             genotype %in% c(&quot;WT&quot;, &quot;YAP1delete&quot;)),
#                         &quot;cer_MET28&quot; = filter(infos_list$cer_TP1,
#                                              genotype %in% c(&quot;WT&quot;, &quot;MET28delete&quot;)),
#                         &quot;par_MET28&quot; = filter(infos_list$par_TP1,
#                                              genotype %in% c(&quot;WT&quot;, &quot;MET28delete&quot;)))
# infos_list_test &lt;- map(infos_list_test, \(x) {
#   x$genotype &lt;- droplevels(x$genotype)
#   return(x)
# })
# counts_list_test &lt;- map(infos_list_test, \(x) {
#   samples &lt;- select(x, sample_name) |&gt; pull()
#   cts &lt;- cbind(counts_tfdel, counts_tfdel_allele)
#   return(cts[, samples])
# })
# 
# #### model fitting ####
# factorizeGenotypeAndTimepoint &lt;- function(.info) {
#   if (length(unique(.info$genotype)) &gt; 1) {
#     .info$genotype &lt;- as.factor(.info$genotype) %&gt;% relevel(ref = &quot;WT&quot;)
#   }
#   if (length(unique(.info$time_point_str)) &gt; 1) {
#     timepoints &lt;- .info$time_point_str %&gt;% unique()
#     reftime &lt;- timepoints[which.min(parse_number(timepoints))] # can&#39;t just do 0 because LowPi has a -5 fml
#     .info$time_point_str &lt;- as.factor(.info$time_point_str) %&gt;% relevel(ref = reftime)
#   }
#   return(.info)
# }
# infos_list &lt;- map(infos_list, factorizeGenotypeAndTimepoint)
# 
# # first exploring detecting which TF deletions a gene is DE in with a positive control
# gene_idx &lt;- &quot;YGR192C&quot; # YGR192C (TDH3) in GCR2 delete
# genedf &lt;- bind_cols(tibble(expr = counts_list$cer_TP1[gene_idx, ]), infos_list$cer_TP1)
# ggplot(genedf, aes(x = genotype, y = expr)) + geom_point() + scale_x_discrete(breaks = NULL)
# modnb &lt;- glm.nb(expr ~ genotype, data = genedf, link = log)
# modpois &lt;- glm(expr ~ genotype, data = genedf, family = poisson(link = &quot;log&quot;))
# summary(modnb)
# genedf$p_hat &lt;- modnb$fitted.values
# plotdf &lt;- pivot_longer(genedf, cols = c(p_hat, expr))
# ggplot(data = plotdf, aes(x = genotype, y = value)) +
#   geom_point(aes(color = name, shape = as.factor(time_point_str))) +
#   scale_x_discrete(breaks = NULL)
# coeffs &lt;- coef(modnb)[grepl(&quot;^genotype&quot;, names(coef(modnb)))]
# pvals &lt;- summary(modnb)$coefficients[,4][grepl(&quot;^genotype&quot;, names(coef(modnb)))]
# sigGenotypes &lt;- names(coeffs[abs(coeffs) &gt; 0.5 &amp; pvals &lt; p_thresh]) %&gt;% gsub(pattern = &quot;genotype&quot;, replacement = &quot;&quot;)
# genedf$sig &lt;- genedf$genotype %in% sigGenotypes
# genedf$sig[genedf$genotype == &quot;WT&quot;] &lt;- NA
# ggplot(data = genedf, aes(x = genotype, y = expr)) +
#   geom_point(aes(color = sig)) +
#   theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1))
# 
# # now making it a function so we can apply it to all genes
# # @input: gene name, counts matrix (where rows are samples, cols are genes),
# #         and info dataframe for LowN experiment in one species
# # @ouput: list of three elements: 1) pvalues of each coefficient (of length nDeletions)
# #                                 2) coefficients (effect sizes) of each TFdel&#39;s effect on the focal gene
# #                                 3) name of which deletion each coefficient/pvalue corresponds to
# 
# getTFdelModResultsByGene &lt;- function(.gene_idx, .cts, .info, .mod_func) {
#   gdf &lt;- bind_cols(tibble(expr = .cts[.gene_idx,]), .info)
#   get1genemod &lt;- function(del) {
#     gdf2gen &lt;- gdf |&gt; filter(genotype %in% c(&quot;WT&quot;, del))
#     gdf2gen$genotype &lt;- droplevels(gdf2gen$genotype)
#     # fitting model
#     m &lt;- tryCatch({
#       .mod_func(.data = gdf2gen)
#     }, error = function(e) {
#       return(NA)
#     })
#     if (!is.na(m)[1]) {
#       pval &lt;- summary(m)$coefficients[,4][paste0(&quot;genotype&quot;, del)]
#       coef &lt;- coef(m)[paste0(&quot;genotype&quot;, del)]
#       return(list(&quot;pval&quot; = as.numeric(pval),
#                   &quot;coef&quot; = as.numeric(coef),
#                   &quot;deletion&quot; = gsub(&quot;delete&quot;, &quot;&quot;, as.character(del))))
#     }
#     else {
#       return(list(&quot;pval&quot; = NA,
#                   &quot;coef&quot; = NA,
#                   &quot;deletion&quot; = NA))
#     }
#   }
#   output &lt;- map_dfr(.x = setdiff(unique(gdf$genotype), &quot;WT&quot;),
#                     .f = get1genemod)
#   output$gene_name &lt;- .gene_idx
#   return(output)
# }
# # models to compare
# mod_poisglm &lt;- function(.data) {
#   m &lt;- glm(expr ~ genotype, data = .data, family = poisson(link = &quot;log&quot;))
#   return(m)
# }
# mod_negbin &lt;- function(.data) {
#   m &lt;- glm.nb(expr ~ genotype, data = .data, link = log)
#   return(m)
# }
# # # tests for getTFdelModResultsByGene
# # getTFdelModResultsByGene(.gene_idx = &quot;YGR192C&quot;,
# #                          .cts = counts_list_test$cer_YAP1,
# #                          .info = infos_list_test$cer_YAP1,
# #                          .mod_func = mod_poisglm)
# 
# # fitting all models (takes 5eva)
# TFdeldfs_pois &lt;- map2(.x = counts_list,
#                       .y = infos_list,
#                       \(x, y) {
#                         output &lt;- map(rownames(x), \(g) {
#                           cat(unique(y$allele), unique(y$time_point_str), 
#                               &quot;pois working on&quot;, g, which(rownames(x) == g), &quot;/&quot;, nrow(x), &quot;\n&quot;)
#                           return(getTFdelModResultsByGene(g, x, y, .mod_func = mod_poisglm))
#                         }) |&gt; purrr::reduce(.f = rbind)
#                       })
# TFdeldfs_negbin &lt;- map2(.x = counts_list,
#                         .y = infos_list,
#                         \(x, y) {
#                           output &lt;- map(rownames(x), \(g) {
#                             cat(unique(y$allele), unique(y$time_point_str), 
#                                 &quot;negbin working on&quot;, g, which(rownames(x) == g), &quot;/&quot;, nrow(x), &quot;\n&quot;)
#                             return(getTFdelModResultsByGene(g, x, y, .mod_func = mod_negbin))
#                           }) |&gt; purrr::reduce(.f = rbind)
#                         })
# 
# # converting natural log fold change to log2 fold change
# # e^x= 2^y
# # log2(e^x) = y
# # x*log2(e) = y
# # x*(1/log(2)) = y
# TFdeldfs_pois &lt;- lapply(TFdeldfs_pois, \(x) {
#   x$coef &lt;- x$coef/log(2)
#   return(x)
# })
# TFdeldfs_negbin &lt;- lapply(TFdeldfs_negbin, \(x) {
#   x$coef &lt;- x$coef/log(2)
#   return(x)
# })
# 
# # standardized mean difference (https://cran.r-project.org/web/packages/TOSTER/vignettes/SMD_calcs.html)
# # a measure of difference btween WT and TFdel to benchmark model effect sizes off of
# # standarized mean difference is calculated as:
# # (m2 - m1)/s_av
# # s_av = sqrt((sd1^2 + sd2^2)/2) (square root of the mean variance)
# # where m1 and sd1 are the mean and sd of counts from WT 
# # and m2 and sd2 are the same for the TF deletion
# # (m2 - m1 allows the sign to match the effect sizes when WT is reference:
# # + means the gene increased expr upon deletion, - means it decreased)
# getStandardizedMeanDiffByGene &lt;- function(.gene_idx, .cts, .info) {
#   cat(&quot;working on&quot;, .gene_idx, 
#       which(.gene_idx == rownames(counts_tfdel)), 
#       &quot;/&quot;, nrow(counts_tfdel), &quot;\n&quot;)
#   gdf &lt;- bind_cols(tibble(expr = .cts[.gene_idx,]), .info)
#   get1geneSMD &lt;- function(del) {
#     gdf2gen &lt;- gdf |&gt; filter(genotype %in% c(&quot;WT&quot;, del))
#     gdf2gen$genotype &lt;- droplevels(gdf2gen$genotype)
#     # calculating SMD
#     wt_vec &lt;- filter(gdf2gen, genotype == &quot;WT&quot;) |&gt; select(expr) |&gt; pull()
#     del_vec &lt;- filter(gdf2gen, genotype != &quot;WT&quot;) |&gt; select(expr) |&gt; pull()
#     m1 &lt;- mean(wt_vec, na.rm = TRUE)
#     sd1 &lt;- sd(wt_vec, na.rm = TRUE)
#     m2 &lt;- mean(del_vec, na.rm = TRUE)
#     sd2 &lt;- sd(del_vec, na.rm = TRUE)
#     s_av &lt;- sqrt((sd1^2 + sd2^2)/2)
#     smd &lt;- (m2-m1)/s_av
#     return(list(&quot;smd&quot; = smd,
#                 &quot;md&quot; = (m2 - m1),
#                 &quot;deletion&quot; = gsub(&quot;delete&quot;, &quot;&quot;, del)))
#   }
#   output &lt;- map_dfr(.x = setdiff(unique(gdf$genotype), &quot;WT&quot;),
#                     .f = get1geneSMD)
#   output$gene_name &lt;- .gene_idx
#   return(output)
# }
# # # tests for getStandardizedMeanDiffByGene
# # # cer
# # getStandardizedMeanDiffByGene(&quot;YGR192C&quot;, counts_list$cer_TP1, infos_list$cer_TP1) |&gt; filter(deletion == &quot;GCR2&quot;)
# 
# # applies getStandardizedMeanDiffByGene to all genes
# getStandardizedMeanDiff &lt;- function(.cts, .info) {
#   output &lt;- map(rownames(.cts), \(g) {
#     getStandardizedMeanDiffByGene(.gene_idx = g,
#                                   .cts = .cts,
#                                   .info = .info)
#   }) |&gt; purrr::reduce(.f = bind_rows)
#   return(output)
# }
# 
# SMDs &lt;- map2(counts_list, infos_list,
#                   .f = getStandardizedMeanDiff)
# 
# ### making dataframes
# load(&quot;data_files/TFdel_DESeq2.RData&quot;)
# 
# # Exploring both of these and DESeq2 results to determine if
# # differences in power btwn Scer and Spar are not seen in every model
# # (and how they relate to &quot;ground-truth&quot; of SMD)
# poisdf &lt;- map2(TFdeldfs_pois,
#                names(TFdeldfs_pois),
#                \(x, y) {
#                  x$organism &lt;- gsub(y, pattern = &quot;_TP[0-9]&quot;,
#                                     replacement = &quot;&quot;)
#                  x$timepoint &lt;- gsub(y, pattern = &quot;[cer_par_]&quot;,
#                                      replacement = &quot;&quot;)
#                  return(x)
#                }) |&gt; purrr::reduce(bind_rows) |&gt; 
#   rename(c(&quot;coef_pois&quot;=&quot;coef&quot;, &quot;pval_pois&quot;=&quot;pval&quot;))
# 
# negbindf &lt;- map2(TFdeldfs_negbin,
#                  names(TFdeldfs_negbin),
#                  \(x, y) {
#                    x$organism &lt;- gsub(y, pattern = &quot;_TP[0-9]&quot;,
#                                       replacement = &quot;&quot;)
#                    x$timepoint &lt;- gsub(y, pattern = &quot;[cer_par_]&quot;,
#                                        replacement = &quot;&quot;)
#                    return(x)
#                  }) |&gt; purrr::reduce(bind_rows) |&gt; 
#   rename(c(&quot;coef_negbin&quot;=&quot;coef&quot;, &quot;pval_negbin&quot;=&quot;pval&quot;))
# 
# SMDdf &lt;- map2(SMDs,
#               names(SMDs),
#               \(x, y) {
#                 x$organism &lt;- gsub(y, pattern = &quot;_TP[0-9]&quot;,
#                                    replacement = &quot;&quot;)
#                 x$timepoint &lt;- gsub(y, pattern = &quot;[cer_par_hyc_hyp_]&quot;,
#                                     replacement = &quot;&quot;)
#                 return(x)
#               }) |&gt; purrr::reduce(bind_rows)
# 
# qcdf &lt;- left_join(poisdf,
#                   negbindf,
#                   by = c(&quot;gene_name&quot;, &quot;organism&quot;, &quot;deletion&quot;, &quot;timepoint&quot;)) |&gt; 
#   left_join(SMDdf, by = c(&quot;gene_name&quot;, &quot;organism&quot;, &quot;deletion&quot;, &quot;timepoint&quot;)) |&gt; 
#   left_join(y = TFdeldf, by = c(&quot;gene_name&quot;, &quot;organism&quot;, &quot;deletion&quot;, &quot;timepoint&quot;))
# 
# # Filtering for genes with high enough expression in LowN
# load(&quot;data_files/FinalDataframe3Disp.RData&quot;)
# qcdf &lt;- filter(qcdf, gene_name %in% unlist(finaldf[finaldf$experiment == &quot;LowN&quot;,
#                                                    &quot;gene_name&quot;]))
# #### QC Exploration ####
# 
# # are effect size estimates and SMD correlated?
# # negbin and SMD
# ggplot(slice_sample(filter(qcdf, pval_negbin &lt; p_thresh), n = 10000),
#        aes(x = smd, y = coef_negbin)) +
#   geom_point(aes(color = coef_negbin &gt; 0)) # certainly none go in opposite directions
# sum(sign(qcdf$smd) != sign(qcdf$coef_negbin) &amp; qcdf$pval_negbin &lt; p_thresh, na.rm = TRUE)
# # pois and SMD
# ggplot(slice_sample(filter(qcdf, pval_pois &lt; p_thresh), n = 10000),
#        aes(x = smd, y = coef_pois)) +
#   geom_point(aes(color = coef_pois &gt; 0)) # perhaps more correlated
# sum(sign(qcdf$smd) != sign(qcdf$coef_pois) &amp; qcdf$pval_pois &lt; p_thresh, na.rm = TRUE)
# # negbin and pois
# ggplot(slice_sample(filter(qcdf, pval_pois &lt; p_thresh), n = 10000),
#        aes(x = coef_negbin, y = coef_pois)) +
#   geom_point() # nevermind, they&#39;re the same
# identical_idxs &lt;- map2(qcdf$coef_pois, qcdf$coef_negbin, identical) |&gt; unlist() 
# sum(identical_idxs) # not exact, but close. Probably stochasticity around the negbin iterative algorithm
# # SMD and DESeq2
# ggplot(slice_sample(filter(qcdf, padj &lt; p_thresh), n = 10000),
#        aes(x = smd, y = lfc)) +
#   geom_point(aes(color = lfc &gt; 0))
# # pois and DESeq2
# ggplot(filter(qcdf, padj &lt; 0.05),
#        aes(x = coef_pois, y = lfc)) +
#   geom_point(aes(color = lfc &gt; 0)) 
# # mostly very related, but there&#39;s a group with very strong 
# # negative effects that pois weren&#39;t calling DE:
# source(&quot;functions_for_figure_scripts.R&quot;)
# load(&quot;data_files/Cleaned_Counts.RData&quot;)
# load(&quot;data_files/Cleaned_Counts_Allele.RData&quot;)
# v_neg_genes &lt;- filter(qcdf, ((padj &lt; 0.05) | (pval_pois &lt; p_thresh)) &amp; 
#                         coef_pois &lt; -20)
# v_neg_genes$deletion |&gt; table() |&gt; sort(decreasing = TRUE)
# v_neg_genes$timepoint |&gt; table() |&gt; sort(decreasing = TRUE)
# v_neg_genes$gene_name |&gt; table() |&gt; sort(decreasing = TRUE)
# v_neg_genes |&gt; filter(deletion == &quot;GAT1&quot;)
# counts_tfdel[&quot;YHR096C&quot;, sample_info_tfdel$organism == &quot;par&quot; &amp;
#                sample_info_tfdel$genotype == &quot;SUM1delete&quot; &amp;
#                sample_info_tfdel$time_point_str == &quot;16 h, low N&quot;] # HXT5, there are 2 replicates --- just both 0
# plotGenesTFdel(.gene_idxs = &quot;YHR096C&quot;, .tf = &quot;SUM1&quot;) # This is real. SUM1 delete shuts off this gene in Spar
# plotGenesTFdel(.gene_idxs = &quot;YHR096C&quot;, .tf = &quot;GAT1&quot;)
# counts_tfdel[&quot;YBR145W&quot;, sample_info_tfdel$organism == &quot;cer&quot; &amp;
#                sample_info_tfdel$genotype == &quot;GCN4delete&quot; &amp;
#                sample_info_tfdel$time_point_str == &quot;16 h, low N&quot;]
# counts_tfdel[&quot;YBL043W&quot;, sample_info_tfdel$organism == &quot;cer&quot; &amp;
#                sample_info_tfdel$genotype == &quot;GCN4delete&quot; &amp;
#                sample_info_tfdel$time_point_str == &quot;16 h, low N&quot;]
# plotGenesTFdel(.gene_idxs = &quot;YBR145W&quot;, .tf = &quot;GCN4&quot;)
# plotGenesTFdel(.gene_idxs = &quot;YBL043W&quot;, .tf = &quot;GCN4&quot;) # ADH5 and ECM13
# # conclusion: these appear to be true effect sizes: these genes are shut off by these TF deletions
# 
# ### Case examples
# # Are there differences in coef/pval for Scer vs Spar?
# # MET28 TP1, no DE genes in Scer
# # pois, pvals not adjusted for multiple testing
# plotdf &lt;- filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot;)
# ggplot(plotdf, aes(x = coef_pois, y = -log10(pval_pois))) +
#   geom_point(aes(color = pval_pois &lt; p_thresh)) +
#   xlim(c(-10, 10))
# plotdf &lt;- filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot;)
# ggplot(plotdf, aes(x = coef_pois, y = -log10(pval_pois))) +
#   geom_point(aes(color = pval_pois &lt; p_thresh)) +
#   xlim(c(-10, 10)) # magnitudes are genuniely smaller for Scer
# filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot; &amp;
#          pval_pois &lt; p_thresh &amp; abs(coef_pois) &gt; coef_thresh) |&gt; nrow()
# filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot; &amp;
#          pval_pois &lt; p_thresh &amp; abs(coef_pois) &gt; coef_thresh) |&gt; nrow()
# # DESeq2
# plotdf &lt;- filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot;)
# ggplot(plotdf, aes(x = lfc, y = -log10(padj))) +
#   geom_point(aes(color = padj &lt; p_thresh)) +
#   xlim(c(-10, 10)) +
#   ylim(c(0, 30))
# plotdf &lt;- filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot;)
# ggplot(plotdf, aes(x = lfc, y = -log10(padj))) +
#   geom_point(aes(color = padj &lt; p_thresh)) +
#   xlim(c(-10, 10)) +
#   ylim(c(0, 30))
# filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot; &amp;
#          padj &lt; p_thresh &amp; abs(lfc) &gt; coef_thresh) |&gt; nrow()
# filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot; &amp;
#          padj &lt; p_thresh &amp; abs(lfc) &gt; coef_thresh) |&gt; nrow()
# 
# # YAP1 TP3, few DE genes in Spar
# # pois, pvals not adjusted for multiple testing
# plotdf &lt;- filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot;)
# ggplot(plotdf, aes(x = coef_pois, y = -log10(pval_pois))) +
#   geom_point(aes(color = pval_pois &lt; p_thresh)) +
#   xlim(c(-10, 10))
# plotdf &lt;- filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot;)
# ggplot(plotdf, aes(x = coef_pois, y = -log10(pval_pois))) +
#   geom_point(aes(color = pval_pois &lt; p_thresh)) +
#   xlim(c(-10, 10)) # I don&#39;t see a diference
# filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot; &amp;
#          pval_pois &lt; p_thresh &amp; abs(coef_pois) &gt; coef_thresh) |&gt; nrow()
# filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;YAP1&quot; &amp;  timepoint == &quot;TP3&quot; &amp;
#          pval_pois &lt; p_thresh &amp; abs(coef_pois) &gt; coef_thresh) |&gt; nrow() # but there is a difference, Spar has 1/4 as many
# # DESeq2
# plotdf &lt;- filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot;)
# ggplot(plotdf, aes(x = lfc, y = -log10(padj))) +
#   geom_point(aes(color = padj &lt; p_thresh)) +
#   xlim(c(-10, 10)) +
#   ylim(c(0, 30))
# plotdf &lt;- filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot;)
# ggplot(plotdf, aes(x = lfc, y = -log10(padj))) +
#   geom_point(aes(color = padj &lt; p_thresh)) +
#   xlim(c(-10, 10)) +
#   ylim(c(0, 30))
# filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;YAP1&quot; &amp;
#          padj &lt; p_thresh &amp; abs(lfc) &gt; coef_thresh) |&gt; nrow()
# filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;YAP1&quot; &amp;
#          padj &lt; p_thresh &amp; abs(lfc) &gt; coef_thresh) |&gt; nrow()
# 
# # How do these vals compare to &quot;gold standard&quot; SMD?
# # Are there more high SMD values for the species with more DE Genes?
# ggplot(filter(qcdf, (deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot;) |
#                 (deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot;)), aes(x = interaction(organism, deletion),
#                    y = smd)) +
#   geom_violin(aes(fill = smd &lt; 0))
# filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot; &amp;
#          abs(smd) &gt; 3) |&gt; nrow()
# filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot; &amp;
#          abs(smd) &gt; 3) |&gt; nrow()
# filter(qcdf, organism == &quot;cer&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot; &amp;
#          abs(smd) &gt; 3) |&gt; nrow()
# filter(qcdf, organism == &quot;par&quot; &amp; deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot; &amp;
#          abs(smd) &gt; 3) |&gt; nrow()
# # There are indeed smaller SMDs in the organism without DE genes
# 
# # Mean difference Scer vs Spar
# # MD is related to expression level, so these will have to be
# # paired observations for each gene in Scer and Spar
# # MET28
# plotdf &lt;- filter(qcdf, deletion == &quot;MET28&quot; &amp; timepoint == &quot;TP1&quot;) |&gt; 
#   pivot_wider(id_cols = &quot;gene_name&quot;, names_from = &quot;organism&quot;, values_from = &quot;md&quot;)
# ggplot(plotdf, aes(x = log2(abs(cer) + 1),
#                    y = log2(abs(par) + 1))) + 
#   geom_hex() +
#   geom_abline(slope = 1, intercept = 0, color = &quot;red&quot;) # Spar has higher (magnitude) MDs
# sum(abs(plotdf$cer) &gt; abs(plotdf$par))
# sum(abs(plotdf$cer) &lt; abs(plotdf$par))
# # YAP1
# plotdf &lt;- filter(qcdf, deletion == &quot;YAP1&quot; &amp; timepoint == &quot;TP3&quot;) |&gt; 
#   pivot_wider(id_cols = &quot;gene_name&quot;, names_from = &quot;organism&quot;, values_from = &quot;md&quot;)
# ggplot(plotdf, aes(x = log2(abs(cer) + 1),
#                    y = log2(abs(par) + 1))) + 
#   geom_hex() +
#   geom_abline(slope = 1, intercept = 0, color = &quot;red&quot;) # Scer has higher (magnitude MDs)
# sum(abs(plotdf$cer) &gt; abs(plotdf$par))
# sum(abs(plotdf$cer) &lt; abs(plotdf$par))
# 
# #### Supplemental figure: TFdel power QC ####
# # TODO: supplemental plot once you&#39;ve run all deletions/organisms/timepoints
# # showing that the TFdeldf, pois, glm.nb, and SMD counts for numbers of DE genes
# # Show the same divergence in nDE genes between species
# 
# # DESeq2 and glm pois make largely similar estimates of fold change
# library(ggExtra)
# plotdf &lt;- filter(drop_na(qcdf), (pval_pois &lt; p_thresh) | (padj &lt; 0.05))
# p &lt;- ggplot(plotdf, 
#        aes(x = coef_pois, y = lfc)) + 
#   geom_point(aes(color = if_else(padj &lt; 0.05,
#                                  true = if_else(pval_pois &lt; p_thresh,
#                                                 true = &quot;sig both&quot;,
#                                                 false = &quot;sig DESeq2&quot;),
#                                  false = if_else(pval_pois &lt; p_thresh,
#                                                  true = &quot;sig glm&quot;,
#                                                  false = &quot;nonsig both&quot;)))) +
#   theme(legend.title = element_blank(),
#         legend.position = &quot;bottom&quot;) +
#   ggtitle(paste0(&quot;n = &quot;, nrow(plotdf)))
# pdf(&quot;../../aligning_the_molecular_phenotype/paper_figures/Supplement/glmVsDESeq2.pdf&quot;,
#     width = 7, height = 7)
# ggMarginal(p, groupColour = TRUE, groupFill = TRUE)
# dev.off()
# 
# # cluster with very negative fold change that&#39;s only sig in DESeq2 
# # is of genes with fully 0 counts upon TFdel:
# table(v_neg_genes$organism) # note they happen to only be parental observations
# plotdf &lt;- map(1:nrow(v_neg_genes), \(i) {
#   org &lt;- v_neg_genes$organism[i]
#   tfdel &lt;- v_neg_genes$deletion[i]
#   g &lt;- v_neg_genes$gene_name[i]
#   tp &lt;- v_neg_genes$timepoint[i]
#   tp_str &lt;- if_else(tp == &quot;TP1&quot;,
#                     true = &quot;0 h, YPD&quot;,
#                     false = if_else(tp == &quot;TP2&quot;,
#                                     true = &quot;1 h, low N&quot;,
#                                     false = &quot;16 h, low N&quot;))
#   wt_avg_expr &lt;- counts_tfdel[g,sample_info_tfdel$organism == org &amp;
#                                 sample_info_tfdel$genotype == &quot;WT&quot; &amp;
#                                 sample_info_tfdel$time_point_str == tp_str] |&gt; 
#     mean()
#   outdf &lt;- tibble(gene_name = g,
#                   organism = org,
#                   deletion = tfdel,
#                   timepoint = tp,
#                   wt_avg_expr = wt_avg_expr,
#                   expr = counts_tfdel[g,sample_info_tfdel$organism == org &amp;
#                                         sample_info_tfdel$genotype == paste0(tfdel, &quot;delete&quot;) &amp;
#                                         sample_info_tfdel$time_point_str == tp_str])
#   return(outdf)
# }) |&gt; purrr::reduce(bind_rows)
# plotdf &lt;- left_join(plotdf, v_neg_genes,
#                     by = c(&quot;gene_name&quot;, &quot;organism&quot;, &quot;deletion&quot;, &quot;timepoint&quot;),
#                     relationship = &quot;many-to-one&quot;) |&gt; 
#   pivot_longer(cols = c(&quot;wt_avg_expr&quot;, &quot;expr&quot;), 
#                names_to = &quot;genotype&quot;, values_to = &quot;expr&quot;)
# plotdf$genotype &lt;- if_else(plotdf$genotype == &quot;wt_avg_expr&quot;,
#                            true = &quot;WT&quot;, false = &quot;TFdel&quot;)
# 
# pdf(&quot;../../aligning_the_molecular_phenotype/paper_figures/Supplement/glmVsDESeq2_negGroup.pdf&quot;,
#     width = 3, height = 2.5)
# ggplot(plotdf, aes(x = genotype, y = expr)) +
#   geom_line(aes(group = interaction(deletion, gene_name, timepoint, organism)),
#             color = &quot;#00BA38&quot;) +
#   theme_classic() +
#   scale_x_discrete(breaks = c(&quot;WT&quot;, &quot;TFdel&quot;),
#                    labels = c(&quot;WT&quot;, &quot;TFdel&quot;),
#                    limits = c(&quot;WT&quot;, &quot;TFdel&quot;)) +
#   ggtitle(paste0(&quot;n = &quot;, nrow(unique(select(plotdf, deletion, gene_name, timepoint, organism)))))
# dev.off()
# 
# ### Heatmap of SMD over a threshold instead of nDE genes
# library(ComplexHeatmap)
# library(circlize)
# # Copy-pasted from TFdel figure script
# parent_tf_tab &lt;- sample_info_tfdel |&gt; filter(time_point_str == &quot;0 h, YPD&quot; &amp;
#                                                genotype != &quot;WT&quot;) |&gt; 
#   select(genotype, organism) |&gt; table()
# parent_goodf_tfs_tp1 &lt;- rownames(parent_tf_tab)[apply(parent_tf_tab, 1, \(x) {all(x &gt; 1)})] |&gt; 
#   gsub(pattern = &quot;delete&quot;, replacement = &quot;&quot;)
# parent_goodf_tfs_tp1
# # TFs with 2 replicates at TP3 in cer/par:
# parent_tf_tab &lt;- sample_info_tfdel |&gt; filter(time_point_str == &quot;16 h, low N&quot; &amp;
#                                                genotype != &quot;WT&quot;) |&gt; 
#   select(genotype, organism) |&gt; table()
# parent_goodf_tfs_tp3 &lt;- rownames(parent_tf_tab)[apply(parent_tf_tab, 1, \(x) {all(x &gt; 1)})] |&gt; 
#   gsub(pattern = &quot;delete&quot;, replacement = &quot;&quot;)
# parent_goodf_tfs_tp3
# # TFs with 2 replicates at TP1 in hyc/hyp:
# hybrid_tf_tab &lt;- sample_info_tfdel_allele |&gt; filter(time_point_str == &quot;0 h, YPD&quot; &amp;
#                                                       genotype != &quot;WT&quot;) |&gt; 
#   select(genotype, allele) |&gt; table()
# hybrid_goodf_tfs_tp1 &lt;- rownames(hybrid_tf_tab)[apply(hybrid_tf_tab, 1, \(x) {all(x &gt; 1)})] |&gt; 
#   gsub(pattern = &quot;delete&quot;, replacement = &quot;&quot;)
# hybrid_goodf_tfs_tp1
# # TFs with 2 replicates at TP2 in hyc/hyp:
# hybrid_tf_tab &lt;- sample_info_tfdel_allele |&gt; filter(time_point_str == &quot;1 h, low N&quot; &amp;
#                                                       genotype != &quot;WT&quot;) |&gt; 
#   select(genotype, allele) |&gt; table()
# hybrid_goodf_tfs_tp2 &lt;- rownames(hybrid_tf_tab)[apply(hybrid_tf_tab, 1, \(x) {all(x &gt; 1)})] |&gt; 
#   gsub(pattern = &quot;delete&quot;, replacement = &quot;&quot;)
# hybrid_goodf_tfs_tp2
# # TFs with 2 replicates at TP3 in hyc/hyp:
# hybrid_tf_tab &lt;- sample_info_tfdel_allele |&gt; filter(time_point_str == &quot;16 h, low N&quot; &amp;
#                                                       genotype != &quot;WT&quot;) |&gt; 
#   select(genotype, allele) |&gt; table()
# hybrid_goodf_tfs_tp3 &lt;- rownames(hybrid_tf_tab)[apply(hybrid_tf_tab, 1, \(x) {all(x &gt; 1)})] |&gt; 
#   gsub(pattern = &quot;delete&quot;, replacement = &quot;&quot;)
# hybrid_goodf_tfs_tp3
# # TFs with 2 replicates in parents for both timepoints:
# goodTFs &lt;- intersect(parent_goodf_tfs_tp1, parent_goodf_tfs_tp3)
# goodTFs
# # rows missing from hybrid TP1:
# setdiff(goodTFs, hybrid_goodf_tfs_tp1)
# # rows missing from hybrid TP3:
# setdiff(goodTFs, hybrid_goodf_tfs_tp3)
# 
# effectsdf &lt;- SMDdf |&gt; 
#   filter(abs(smd) &gt; 3 &amp; 
#            timepoint != &quot;TP2&quot; &amp;
#            deletion %in% goodTFs) |&gt; 
#   group_by(deletion, organism, timepoint) |&gt; 
#   summarise(nGenes = n()) |&gt; 
#   pivot_wider(id_cols = c(&quot;deletion&quot;), 
#               names_from = c(&quot;organism&quot;, &quot;timepoint&quot;), 
#               values_from = &quot;nGenes&quot;) |&gt; 
#   ungroup()
# effects_mat &lt;- select(effectsdf, -deletion) |&gt; as.matrix()
# rownames(effects_mat) &lt;- effectsdf$deletion
# effects_mat[is.na(effects_mat)] &lt;- 0 # because we already eliminated TFs without replicates, any NAs come from missing combinations in group_by
# # adding back NAs for missing TFs in the hybrid
# # TP1
# effects_mat[rownames(effects_mat) %in% setdiff(goodTFs, hybrid_goodf_tfs_tp1), 
#             colnames(effects_mat) %in% c(&quot;hyc_TP1&quot;, &quot;hyp_TP1&quot;)] &lt;- NA
# # # TP2
# # effects_mat[rownames(effects_mat) %in% setdiff(goodTFs, hybrid_goodf_tfs_tp2), 
# #             colnames(effects_mat) %in% c(&quot;hyc_TP2&quot;, &quot;hyp_TP2&quot;)] &lt;- NA
# # TP3
# effects_mat[rownames(effects_mat) %in% setdiff(goodTFs, hybrid_goodf_tfs_tp3), 
#             colnames(effects_mat) %in% c(&quot;hyc_TP3&quot;, &quot;hyp_TP3&quot;)] &lt;- NA
# 
# col_fun &lt;- colorRamp2(c(0, 50, 300), c(&quot;blue&quot;, &quot;yellow&quot;, &quot;red&quot;))
# p &lt;- Heatmap(effects_mat, col = col_fun, na_col = &quot;grey80&quot;,
#              column_order = c(&quot;cer_TP1&quot;, &quot;par_TP1&quot;, &quot;cer_TP3&quot;, &quot;par_TP3&quot;, 
#                               &quot;hyc_TP1&quot;, &quot;hyp_TP1&quot;, &quot;hyc_TP3&quot;, &quot;hyp_TP3&quot;),
#              # row_order = tf_order,
#              cell_fun = function(j, i, x, y, width, height, fill) {
#                output &lt;- if_else(!(is.na(effects_mat[i, j])), 
#                                  true = as.character(effects_mat[i, j]), 
#                                  false = &quot;-&quot;)
#                grid.text(output, x, y, gp = gpar(fontsize = 10))
#              })
# 
# pdf(&quot;../../aligning_the_molecular_phenotype/paper_figures/Supplement/TFdel_SMD_heatmap.pdf&quot;, 
#     width = 5, height = 8)
# p
# dev.off()
# 
# # TODO: DESeq2 seems to have many more significant negative
# # log2 fold change effects than positive ones, 
# # Is this also true for SMD of a large enough magnitude?
# 
# # saving
# save(TFdeldfs_pois,
#      TFdeldfs_negbin,
#      SMDs, SMDdf, poisdf, 
#      qcdf, file = &quot;data_files/QC_TFdel.RData&quot;)
# load(file = &quot;data_files/QC_TFdel.RData&quot;)
# 
# #### effect size distributions by TF ####
# # do any TFs have more nsig than others within each dataset?
# plotlist &lt;- vector(mode = &quot;list&quot;, length = length(TFdeldfs_pois))
# for (i in c(1:length(TFdeldfs_pois))) {
#   df &lt;- TFdeldfs_pois[[i]]
#   plotdf &lt;- df |&gt; filter(abs(coef) &gt; 0.5 &amp; pval &lt; p_thresh) |&gt; 
#     group_by(deletion)
#   p &lt;- ggplot(plotdf, aes(x = deletion)) +
#     geom_bar(aes(fill = deletion)) +
#     ggtitle(names(TFdeldfs_pois)[i]) +
#     theme(legend.position = &quot;none&quot;, 
#           axis.text.x = element_text(angle = 90)) +
#     ylim(c(0, 1600))
#   plotlist[[i]] &lt;- p
# }
# ggarrange(plotlist = plotlist, ncol = 4, nrow = 2)
# 
# # both barkai and redhuis have the discrepancy of par having ~2x as many DE as cer
# # hybrids don&#39;t, so this isn&#39;t an alignment issue
# # no obvious single TFs that are driving the pattern
# # barkai vs redhuis alingment have slight differences in power, 
# # but are mostly similar, especially hybrid 
# # (except TEC1, which has more DE in redhuis hyc and hyp)
# 
# # but at some point, I didn&#39;t have this issue
# 
# #### comparing models ####
# 
# # overall n DE gene/TF combinations
# # cer pois, no interaction
# sum(TFdeldfs_pois$redhuisdf_cer$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_pois$redhuisdf_cer$coef) &gt; 0.25)
# # par pois, no interaction
# sum(TFdeldfs_pois$redhuisdf_par$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_pois$redhuisdf_par$coef) &gt; 0.25) # wait that&#39;s not that many more
# # cer pois, with interaction
# sum(TFdeldfs_poisint$redhuisdf_cer$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_poisint$redhuisdf_cer$coef) &gt; 0.25)
# # par pois, with interaction
# sum(TFdeldfs_poisint$redhuisdf_par$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_poisint$redhuisdf_par$coef) &gt; 0.25)
# # extreme values only
# # cer pois, no interaction
# sum(TFdeldfs_pois$redhuisdf_cer$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_pois$redhuisdf_cer$coef) &gt; 2)
# # par pois, no interaction
# sum(TFdeldfs_pois$redhuisdf_par$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_pois$redhuisdf_par$coef) &gt; 2) # ah now it&#39;s over twice as many
# # cer pois, with interaction
# sum(TFdeldfs_poisint$redhuisdf_cer$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_poisint$redhuisdf_cer$coef) &gt; 2)
# # par pois, with interaction
# sum(TFdeldfs_poisint$redhuisdf_par$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_poisint$redhuisdf_par$coef) &gt; 2)
# # cer negbin, no interaction
# sum(TFdeldfs_negbin$redhuisdf_cer$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_negbin$redhuisdf_cer$coef) &gt; 2)
# # par negbin, no interaction
# sum(TFdeldfs_negbin$redhuisdf_par$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_negbin$redhuisdf_par$coef) &gt; 2, na.rm = TRUE)
# # cer negbin, with interaction
# sum(TFdeldfs_negbinint$redhuisdf_cer$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_negbinint$redhuisdf_cer$coef) &gt; 2)
# # par negbin, with interaction
# sum(TFdeldfs_negbinint$redhuisdf_par$pval &lt; p_thresh &amp; 
#       abs(TFdeldfs_negbinint$redhuisdf_par$coef) &gt; 2, na.rm = TRUE)
# 
# # conclusions: no obvious solution, but par seems to have more extreme effect sizes,
# # especially in poisson models and models without interaction terms
# 
# ### density plots of effect sizes in cer versus par
# # SMD plot for ground-truth reference
# # all SMDs, regardless of size
# plotdf &lt;- SMDdf |&gt; pivot_longer(cols = c(&quot;cer&quot;, &quot;par&quot;),
#                                 names_to = &quot;species&quot;,
#                                 values_to = &quot;smd&quot;) 
# ggplot(plotdf, aes(x = smd)) + 
#   geom_density(aes(fill = species), alpha = 0.5) # par has stronger effects in both directions, not just lower tail
# 
# # now models
# # pois, no interaction
# plotdf &lt;- TFdeldfs_pois$redhuisdf_cer |&gt; 
#   #filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;cer&quot;)
# plotdf &lt;- TFdeldfs_pois$redhuisdf_par |&gt; 
#   #filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;par&quot;) |&gt; 
#   bind_rows(y = plotdf)
# ggplot(plotdf, aes(x = (-log(abs(coef)))*sign(coef))) + 
#   geom_density(aes(fill = species), alpha = 0.5) +
#   geom_vline(xintercept = 1) +
#   geom_vline(xintercept = -1)
# # okayyy there is a bias for paradoxus to have more negative strong effect sizes
# # this becomes overwhelming for stronger than effect sizes less than -1:
# ggplot(plotdf, aes(x = coef)) + 
#   geom_density(aes(fill = species), alpha = 0.5) +
#   geom_vline(xintercept = 1) +
#   geom_vline(xintercept = -1) +
#   ylim(c(0, 0.1)) + xlim(c(-5, 5))
# 
# # pois, with interaction
# plotdf &lt;- TFdeldfs_poisint$redhuisdf_cer |&gt; 
#   filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;cer&quot;)
# plotdf &lt;- TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;par&quot;) |&gt; 
#   bind_rows(y = plotdf)
# ggplot(plotdf, aes(x = coef)) + 
#   geom_density(aes(fill = species), alpha = 0.5) +
#   geom_vline(xintercept = 1) +
#   geom_vline(xintercept = -1)
# 
# # negbin, no interaction
# plotdf &lt;- TFdeldfs_negbin$redhuisdf_cer |&gt; 
#   filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;cer&quot;)
# plotdf &lt;- TFdeldfs_negbin$redhuisdf_par |&gt; 
#   filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;par&quot;) |&gt; 
#   bind_rows(y = plotdf)
# ggplot(plotdf, aes(x = coef)) + 
#   geom_density(aes(fill = species), alpha = 0.5) +
#   geom_vline(xintercept = 1) +
#   geom_vline(xintercept = -1)
# 
# # negbin, with interaction
# plotdf &lt;- TFdeldfs_negbinint$redhuisdf_cer |&gt; 
#   filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;cer&quot;)
# plotdf &lt;- TFdeldfs_negbinint$redhuisdf_par |&gt; 
#   filter(pval &lt; p_thresh) |&gt; 
#   select(gene_name, deletion, coef) |&gt; 
#   mutate(species = &quot;par&quot;) |&gt; 
#   bind_rows(y = plotdf)
# ggplot(plotdf, aes(x = coef)) + 
#   geom_density(aes(fill = species), alpha = 0.5) +
#   geom_vline(xintercept = 1) +
#   geom_vline(xintercept = -1)
# 
# 
# #### finding a better coef thresh cutoff ####
# 
# # seems like I&#39;m being too stringent with my effect size cutoff
# # use SMD of ~1 to figure out what the corresponding coef_thresh would be
# # and use that instead
# # (or look back at those coef vs smd plots and see where coef starts to &quot;fan out&quot;
# # i.e. becomes unrelated to smd)
# 
# # control: no accounting for time, should be equivalent to smd
# plotdf &lt;- left_join(SMDdf, 
#                     rename(TFdeldfs_notime$redhuisdf_cer,
#                            c(&quot;coef_cer&quot;=&quot;coef&quot;,
#                              &quot;pval_cer&quot;=&quot;pval&quot;)),
#                     by = join_by(gene_name, deletion)) |&gt; 
#   left_join(y = rename(TFdeldfs_notime$redhuisdf_par,
#                        c(&quot;coef_par&quot;=&quot;coef&quot;,
#                          &quot;pval_par&quot;=&quot;pval&quot;)),
#             by = join_by(gene_name, deletion))
# 
# ggplot(slice_sample(plotdf, n = 10000), 
#        aes(x = cer,
#            y = coef_cer)) +
#   geom_point(aes(color = pval_cer &lt; p_thresh)) +
#   geom_vline(xintercept = c(-1, 1))
# 
# ### smd versus coef in pois with interaction (most realistic model)
# 
# plotdf &lt;- left_join(SMDdf, 
#                     rename(TFdeldfs_poisint$redhuisdf_cer,
#                            c(&quot;coef_cer&quot;=&quot;coef&quot;,
#                              &quot;pval_cer&quot;=&quot;pval&quot;)),
#                     by = join_by(gene_name, deletion)) |&gt; 
#   left_join(y = rename(TFdeldfs_poisint$redhuisdf_par,
#                        c(&quot;coef_par&quot;=&quot;coef&quot;,
#                          &quot;pval_par&quot;=&quot;pval&quot;)),
#             by = join_by(gene_name, deletion))
# 
# # cer
# ggplot(slice_sample(plotdf, n = 10000), 
#        aes(x = cer,
#            y = coef_cer)) +
#   geom_point(aes(color = pval_cer &lt; p_thresh)) +
#   geom_vline(xintercept = c(-1, 1)) + ylim(c(-5, 5))
# 
# # par
# ggplot(slice_sample(plotdf, n = 10000), 
#        aes(x = par,
#            y = coef_par)) +
#   geom_point(aes(color = pval_par &lt; p_thresh)) +
#   geom_vline(xintercept = c(-1, 1)) + ylim(c(-5, 5))
# 
# 
# #### barplots ####
# coef_thresh &lt;- 0.5
# load(&quot;data_files/modules.RData&quot;)
# plotTFBarplots &lt;- function(.deldf_cer, .deldf_par, .m_genedf,
#                            .useLiteralColors = TRUE) {
#   common_TFs &lt;- intersect(unique(.deldf_cer$deletion), unique(.deldf_par$deletion))
#   plotdf_cer &lt;-  .deldf_cer |&gt; left_join(y = select(.m_genedf, gene_name, CCM_color),
#                                          by = &quot;gene_name&quot;)
#   plotdf_cer$sig &lt;- apply(plotdf_cer, 1, \(x) {
#     coeff &lt;- x[&quot;coef&quot;] |&gt; as.numeric()
#     pvalue &lt;- x[&quot;pval&quot;] |&gt; as.numeric()
#     if (coeff &gt; coef_thresh &amp; pvalue &lt; p_thresh) {
#       return(&quot;up&quot;)
#     }
#     if (coeff &lt; -coef_thresh &amp; pvalue &lt; p_thresh) {
#       return(&quot;down&quot;)
#     }
#     else {
#       return(&quot;none&quot;)
#     }
#   })
#   # repeat for paradoxus
#   plotdf_par &lt;-  .deldf_par |&gt; left_join(y = select(.m_genedf, gene_name, CCM_color),
#                                          by = &quot;gene_name&quot;)
#   plotdf_par$sig &lt;- apply(plotdf_par, 1, \(x) {
#     coeff &lt;- x[&quot;coef&quot;] |&gt; as.numeric()
#     pvalue &lt;- x[&quot;pval&quot;] |&gt; as.numeric()
#     if (coeff &gt; coef_thresh &amp; pvalue &lt; p_thresh) {
#       return(&quot;up&quot;) 
#     }
#     if (coeff &lt; -coef_thresh &amp; pvalue &lt; p_thresh) {
#       return(&quot;down&quot;)
#     }
#     else {
#       return(&quot;none&quot;)
#     }
#   })
#   # separating genes found in both species from genes that are species unique
#   plotdf_common &lt;- bind_rows(select(plotdf_cer, gene_name, CCM_color, deletion, sig), 
#                              select(plotdf_par, gene_name, CCM_color, deletion, sig)) |&gt; filter(sig != &quot;none&quot;)
#   plotdf_common &lt;- plotdf_common[duplicated(plotdf_common),] |&gt; 
#     mutate(idx =  paste0(gene_name, deletion))
#   # filtering out these gene/TF combinations in cer and par
#   plotdf_cer &lt;- plotdf_cer |&gt; filter(sig != &quot;none&quot;) |&gt; 
#     mutate(idx = paste0(gene_name, deletion)) |&gt;
#     filter(!(idx %in% plotdf_common$idx))
#   plotdf_par &lt;- plotdf_par |&gt; filter(sig != &quot;none&quot;) |&gt; 
#     mutate(idx = paste0(gene_name, deletion)) |&gt;
#     filter(!(idx %in% plotdf_common$idx))
#   if (.useLiteralColors) {
#     plotdf_cer$CCM_color &lt;- factor(plotdf_cer$CCM_color) |&gt; relevel(ref = &quot;none&quot;)
#     plotdf_par$CCM_color &lt;- factor(plotdf_par$CCM_color) |&gt; relevel(ref = &quot;none&quot;)
#     plotdf_common$CCM_color &lt;- factor(plotdf_common$CCM_color) |&gt; relevel(ref = &quot;none&quot;)
#   }
#   # counting
#   plotdf_cer &lt;- plotdf_cer |&gt; 
#     group_by(CCM_color, deletion) |&gt; 
#     summarise(up = sum(sig == &quot;up&quot;), down = -sum(sig == &quot;down&quot;))
#   plotdf_par &lt;- plotdf_par |&gt; 
#     group_by(CCM_color, deletion) |&gt; 
#     summarise(up = sum(sig == &quot;up&quot;), down = -sum(sig == &quot;down&quot;))
#   plotdf_common &lt;- plotdf_common |&gt; 
#     group_by(CCM_color, deletion) |&gt; 
#     summarise(up = sum(sig == &quot;up&quot;), down = -sum(sig == &quot;down&quot;))
#   # getting overall max, to ensure equivalent y-axes between plots
#   eachtf &lt;- bind_rows(mutate(plotdf_cer, plot = &quot;cer&quot;),
#                       mutate(plotdf_par, plot = &quot;par&quot;),
#                       mutate(plotdf_common, plot = &quot;common&quot;)) |&gt; 
#     group_by(deletion, plot) |&gt; 
#     summarise(up = sum(up), down = sum(down)) 
#   max_y &lt;- max(abs(c(eachtf$up, eachtf$down)))
#   # plotting
#   # genes with a TF response only in cer
#   p_cer &lt;- plotdf_cer |&gt; 
#     ggplot(aes(x = deletion, y = up, fill = CCM_color)) +
#     geom_col(position = &quot;stack&quot;) +
#     geom_col(aes(y = down), position = &quot;stack&quot;) + 
#     geom_hline(yintercept = 0) +
#     ylab(&quot;&quot;) +
#     xlab(&quot;Transcription Factor Deletion&quot;) +
#     scale_x_discrete(limits = common_TFs, labels = common_TFs) +
#     #theme_classic() + 
#     theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), legend.position = &quot;none&quot;) + 
#     ggtitle(&quot;genes only regulated in S. cereivisiae&quot;) +
#     ylim(c(-max_y - 5, max_y + 5))
#   # genes with a TF response only in par
#   p_par &lt;- plotdf_par |&gt; 
#     ggplot(aes(x = deletion, y = up, fill = CCM_color)) +
#     geom_col(position = &quot;stack&quot;) +
#     geom_col(aes(y = down), position = &quot;stack&quot;) + 
#     geom_hline(yintercept = 0) +
#     ylab(&quot;&quot;) +
#     xlab(&quot;Transcription Factor Deletion&quot;) +
#     scale_x_discrete(limits = common_TFs, labels = common_TFs) +
#     #theme_classic() + 
#     theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), legend.position = &quot;none&quot;) + 
#     ggtitle(&quot;genes only regulated in S. paradoxus&quot;) +
#     ylim(c(-max_y - 5, max_y + 5))
#   # genes with conserved TF responses in cer and par
#   p_common &lt;- plotdf_common |&gt; 
#     ggplot(aes(x = deletion, y = up, fill = CCM_color)) +
#     geom_col(position = &quot;stack&quot;) +
#     geom_col(aes(y = down), position = &quot;stack&quot;) + 
#     geom_hline(yintercept = 0) +
#     ylab(&quot;&quot;) +
#     xlab(&quot;Transcription Factor Deletion&quot;) +
#     scale_x_discrete(limits = common_TFs, labels = common_TFs) +
#     #theme_classic() + 
#     theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), legend.position = &quot;none&quot;) + 
#     ggtitle(&quot;Genes with conserved regulation between species&quot;) +
#     ylim(c(-max_y - 5, max_y + 5))
#   if (.useLiteralColors) {
#     p_cer &lt;- p_cer +
#       scale_fill_discrete(limits = plotdf_cer$CCM_color, 
#                           type = gsub(&quot;none&quot;, &quot;grey80&quot;, plotdf_cer$CCM_color))
#     p_par &lt;- p_par +
#       scale_fill_discrete(limits = plotdf_par$CCM_color, 
#                           type = gsub(&quot;none&quot;, &quot;grey80&quot;, plotdf_par$CCM_color))
#     p_common &lt;- p_common +
#       scale_fill_discrete(limits = plotdf_common$CCM_color, 
#                           type = gsub(&quot;none&quot;, &quot;grey80&quot;, plotdf_common$CCM_color)) 
#   }
#   return(list(cer = p_cer, par = p_par, common = p_common))
# }
# ### parents
# # redhuis alignment, pois with interaction
# barplots_redhuis_parents &lt;- plotTFBarplots(TFdeldfs_poisint$redhuisdf_cer, 
#                                            TFdeldfs_poisint$redhuisdf_par, module_genedf25)
# ggarrange(plotlist = barplots_redhuis_parents, nrow = 3, ncol = 1)
# # barkai alignment, pois with interaction
# barplots_barkai_parents &lt;- plotTFBarplots(TFdeldfs_poisint$barkaidf_cer, 
#                                            TFdeldfs_poisint$barkaidf_par, module_genedf25)
# ggarrange(plotlist = barplots_barkai_parents, nrow = 3, ncol = 1)
# # redhuis alignment, pois no interaction
# barplots_redhuis_parents &lt;- plotTFBarplots(TFdeldfs_pois$redhuisdf_cer, 
#                                            TFdeldfs_pois$redhuisdf_par, module_genedf25)
# ggarrange(plotlist = barplots_redhuis_parents, nrow = 3, ncol = 1)
# # barkai alignment, pois no interaction
# barplots_barkai_parents &lt;- plotTFBarplots(TFdeldfs_pois$barkaidf_cer, 
#                                           TFdeldfs_pois$barkaidf_par, module_genedf25)
# ggarrange(plotlist = barplots_barkai_parents, nrow = 3, ncol = 1)
# ### hybrids
# # redhuis alignment, pois with interaction
# barplots_redhuis_hybrid &lt;- plotTFBarplots(TFdeldfs_poisint$redhuisdf_hyc, 
#                                            TFdeldfs_poisint$redhuisdf_hyp, module_genedf25)
# ggarrange(plotlist = barplots_redhuis_hybrid, nrow = 3, ncol = 1)
# # barkai alignment, pois with interaction
# barplots_barkai_hybrid &lt;- plotTFBarplots(TFdeldfs_poisint$barkaidf_hyc, 
#                                           TFdeldfs_poisint$barkaidf_hyp, module_genedf25)
# ggarrange(plotlist = barplots_barkai_hybrid, nrow = 3, ncol = 1)
# # redhuis alignment, pois no interaction
# barplots_redhuis_hybrid &lt;- plotTFBarplots(TFdeldfs_pois$redhuisdf_hyc, 
#                                            TFdeldfs_pois$redhuisdf_hyp, module_genedf25)
# ggarrange(plotlist = barplots_redhuis_hybrid, nrow = 3, ncol = 1)
# # barkai alignment, pois no interaction
# barplots_barkai_hybrid &lt;- plotTFBarplots(TFdeldfs_pois$barkaidf_hyc, 
#                                           TFdeldfs_pois$barkaidf_hyp, module_genedf25)
# ggarrange(plotlist = barplots_barkai_hybrid, nrow = 3, ncol = 1)
# 
# # conclusions: parents show same pattern regardless of which alignment/poisson formula is used
# # hybrids have consistent DE genes in all modules across all TFs, which does not seem right
# 
# #### Do the hybrids really have DE genes in all modules for all TFs? ####
# 
# # parents
# # pois with interaction, at least one sig gene x TF combo
# # cer sig
# random_gene_tf &lt;- TFdeldfs_poisint$redhuisdf_cer |&gt; 
#   filter(abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   slice_sample(n = 1) |&gt; select(gene_name, deletion)
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts$cer[,random_gene_tf$gene_name]), redhuis_info$cer)
# sigGenotypes &lt;- TFdeldfs_negbin$redhuisdf_cer |&gt; 
#   filter(gene_name == random_gene_tf$gene_name &amp;
#            abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(data = genedf, aes(x = genotype, y = expr)) +
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) +
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# # par sig
# random_gene_tf &lt;- TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   slice_sample(n = 1) |&gt; select(gene_name, deletion)
# TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(gene_name == random_gene_tf$gene_name &amp; 
#            deletion == random_gene_tf$deletion)
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts$par[,random_gene_tf$gene_name]), redhuis_info$par)
# sigGenotypes &lt;- TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(gene_name == random_gene_tf$gene_name &amp;
#            abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(data = genedf, aes(x = genotype, y = expr)) +
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) +
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# 
# # TODO: why does YHL035C have a coef of 1 in AFT1delete, but there are
# # plenty of other genotypes that appear to have the same effect 
# # (ARG81/GLN3/PHD1 for example) that have small or negative effect sizes?
# # TODO: check if notime model helps with this
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts$par[,&quot;YHL035C&quot;]), redhuis_info$par)
# ggplot(filter(genedf, genotype %in% c(&quot;WT&quot;, paste0(c(&quot;AFT1&quot;, &quot;ARG81&quot;, &quot;GLN3&quot;, &quot;PHD1&quot;), &quot;delete&quot;))), 
#        aes(x = genotype, y = expr)) + geom_point(aes(color = time_point_str))
# TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(gene_name == &quot;YHL035C&quot; &amp; 
#            deletion == &quot;AFT1&quot;)
# TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(gene_name == &quot;YHL035C&quot; &amp; 
#            deletion == &quot;ARG81&quot;)
# TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(gene_name == &quot;YHL035C&quot; &amp; 
#            deletion == &quot;GLN3&quot;)
# TFdeldfs_poisint$redhuisdf_par |&gt; 
#   filter(gene_name == &quot;YHL035C&quot; &amp; 
#            deletion == &quot;PHD1&quot;)
# 
# # hybrids
# # pois with interaction, at least one sig gene x TF combo
# # hyc sig
# random_gene_tf &lt;- TFdeldfs_poisint$redhuisdf_hyc |&gt; 
#   filter(abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   slice_sample(n = 1) |&gt; select(gene_name, deletion)
# # hyc plot
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts_allele$cer[,random_gene_tf$gene_name]), redhuis_info_allele$cer)
# sigGenotypes &lt;- TFdeldfs_poisint$redhuisdf_hyc |&gt; 
#   filter(gene_name == random_gene_tf$gene_name &amp;
#            abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(data = genedf, aes(x = genotype, y = expr)) +
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) +
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# # hyp plot
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts_allele$par[,random_gene_tf$gene_name]), redhuis_info_allele$par)
# sigGenotypes &lt;- TFdeldfs_poisint$redhuisdf_hyp |&gt; 
#   filter(gene_name == random_gene_tf$gene_name &amp;
#            abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(data = genedf, aes(x = genotype, y = expr)) +
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) +
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# # hyp sig
# random_gene_tf &lt;- TFdeldfs_poisint$redhuisdf_hyp |&gt; 
#   filter(abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   slice_sample(n = 1) |&gt; select(gene_name, deletion)
# # hyp plot
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts_allele$par[,random_gene_tf$gene_name]), redhuis_info_allele$par)
# sigGenotypes &lt;- TFdeldfs_poisint$redhuisdf_hyp |&gt; 
#   filter(gene_name == random_gene_tf$gene_name &amp;
#            abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(data = genedf, aes(x = genotype, y = expr)) +
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) +
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# # hyc plot
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts_allele$cer[,random_gene_tf$gene_name]), redhuis_info_allele$cer)
# sigGenotypes &lt;- TFdeldfs_poisint$redhuisdf_hyc |&gt; 
#   filter(gene_name == random_gene_tf$gene_name &amp;
#            abs(coef) &gt; 1 &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(data = genedf, aes(x = genotype, y = expr)) +
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) +
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# 
# 
# 
# #### Ensuring Hybrid LFC measures comparable to parents ####
# 
# # main goal: Get a measure of expression change that isn&#39;t sensitive to hybrids
# #            having less variation between measurements than parents
# #            (thus having larger expr change measurements than parents)
# # I think this will be achievable using log fold change,
# # which *I don&#39;t believe* is the same as exp(genotype coefficient) in the glm,
# # which in the simplest case (expr ~ genotype) is just the ratio of
# # means of expression (not the ratio of means of log2(expr)), 
# # which would be comparable between expression levels
# # I *believe* LFC is instead this ratio of log2(expr mutant)/log2(expr WT)
# # but I need to verify that first
# 
# # TDH3 in GCR2delete is a good starting example. If this expr
# # change isn&#39;t the same for hybrids and parents, nothing is
# gene_idx &lt;- &quot;YGR192C&quot;
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts$cer[,gene_idx]), redhuis_info$cer) |&gt; 
#   filter(time_point_str == &quot;16 h, low N&quot;)
# genedf$genotype &lt;- as.factor(genedf$genotype) |&gt; relevel(ref = &quot;WT&quot;)
# sigGenotypes &lt;- TFdeldfs_pois$redhuisdf_cer |&gt; 
#   filter(gene_name == gene_idx &amp; abs(coef) &gt; coef_thresh &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(genedf, aes(x = genotype, y = expr)) + 
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) + 
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# mod &lt;- glm(expr ~ genotype, data = genedf, family = poisson(link = &quot;log&quot;))
# mod$coefficients |&gt; sort(decreasing = TRUE) # a cutoff of 1 would unfortunately still get GCR2, but that might be necessary
# # now tf vs wt for gln3 specifically
# genedf$p_hat &lt;- mod$fitted.values
# gcr2_effect_size &lt;- mod$coefficients[&quot;genotypeGCR2delete&quot;] |&gt; as.numeric()
# slope &lt;- exp(gcr2_effect_size)
# intercept &lt;- mod$coefficients[&quot;(Intercept)&quot;] |&gt; as.numeric()
# plotdf &lt;- pivot_longer(genedf, cols = c(p_hat, expr))
# ggplot(data = plotdf, aes(x = genotype, y = value)) +
#   geom_point(aes(color = name, shape = as.factor(time_point_str))) +
#   scale_x_discrete(breaks = NULL)
# plotdf &lt;- plotdf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
#   pivot_wider(id_cols = c(&quot;time_point_str&quot;, &quot;name&quot;, &quot;well_flask_ID&quot;), 
#               names_from = &quot;genotype&quot;,
#               values_from = &quot;value&quot;)
# ggplot(plotdf, aes(x = WT, y = GCR2delete)) + 
#   geom_point(aes(color = name, shape = time_point_str)) +
#   geom_abline(slope = 1, intercept = 0) +
#   geom_abline(slope = slope, intercept = intercept, color = &quot;gold&quot;) +
#   xlim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE))) +
#   ylim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE)))
# # genedf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
# #   select(expr, p_hat, genotype, well_flask_ID) |&gt; View()
# # mean WT:
# genedf |&gt; filter(genotype == &quot;WT&quot;) |&gt; 
#   select(expr) |&gt; pull() |&gt; mean()
# # mean GCR2 delete:
# genedf |&gt; filter(genotype == &quot;GCR2delete&quot;) |&gt; 
#   select(expr) |&gt; pull() |&gt; mean()
# # sd WT:
# genedf |&gt; filter(genotype == &quot;WT&quot;) |&gt; 
#   select(expr) |&gt; pull() |&gt; sd()
# # sd GCR2 delete:
# genedf |&gt; filter(genotype == &quot;GCR2delete&quot;) |&gt; 
#   select(expr) |&gt; pull() |&gt; sd()
# 
# # effect size, log(mean(TFdel)/mean(WT)):
# gcr2_effect_size
# log(mean(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;])/mean(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# # log2 effect size--- a value around 1 means expr doubled
# log2(mean(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;])/mean(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# log2(exp(gcr2_effect_size))
# # &quot;log-scale&quot; fold change (lsfc, name pending): mean(log2(TFdel))/mean(log2(WT)):
# mean(log2(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;]))/mean(log2(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# # a decrease in expr is anything less than 1 now: lsfc(WT = 2^14, TFdel = 2^13) = 13/14 = 0.9ish
# 
# # repeat with par --- lower expressed and slightly smaller magnitude of expr change
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts$par[,gene_idx]), redhuis_info$par) |&gt; 
#   filter(time_point_str == &quot;16 h, low N&quot;)
# genedf$genotype &lt;- as.factor(genedf$genotype) |&gt; relevel(ref = &quot;WT&quot;)
# sigGenotypes &lt;- TFdeldfs_pois$redhuisdf_par |&gt; 
#   filter(gene_name == gene_idx &amp; abs(coef) &gt; coef_thresh &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(genedf, aes(x = genotype, y = expr)) + 
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) + 
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# mod &lt;- glm(expr ~ genotype, data = genedf, family = poisson(link = &quot;log&quot;))
# mod$coefficients |&gt; sort(decreasing = TRUE) # a cutoff of 1 would unfortunately still get GCR2, but that might be necessary
# # now tf vs wt for gln3 specifically
# genedf$p_hat &lt;- mod$fitted.values
# gcr2_effect_size &lt;- mod$coefficients[&quot;genotypeGCR2delete&quot;] |&gt; as.numeric()
# slope &lt;- exp(gcr2_effect_size)
# intercept &lt;- mod$coefficients[&quot;(Intercept)&quot;] |&gt; as.numeric()
# plotdf &lt;- pivot_longer(genedf, cols = c(p_hat, expr))
# ggplot(data = plotdf, aes(x = genotype, y = value)) +
#   geom_point(aes(color = name, shape = as.factor(time_point_str))) +
#   scale_x_discrete(breaks = NULL)
# plotdf &lt;- plotdf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
#   pivot_wider(id_cols = c(&quot;time_point_str&quot;, &quot;name&quot;, &quot;well_flask_ID&quot;), 
#               names_from = &quot;genotype&quot;,
#               values_from = &quot;value&quot;)
# ggplot(plotdf, aes(x = WT, y = GCR2delete)) + 
#   geom_point(aes(color = name, shape = time_point_str)) +
#   geom_abline(slope = 1, intercept = 0) +
#   geom_abline(slope = slope, intercept = intercept, color = &quot;gold&quot;) +
#   xlim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE))) +
#   ylim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE)))
# # genedf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
# #   select(expr, p_hat, genotype, well_flask_ID) |&gt; View()
# # mean WT:
# genedf |&gt; filter(genotype == &quot;WT&quot;) |&gt; select(expr) |&gt; pull() |&gt; mean()
# # mean GCR2delete:
# genedf |&gt; filter(genotype == &quot;GCR2delete&quot;) |&gt; select(expr) |&gt; pull() |&gt; mean()
# 
# # effect size/LFC, log(mean(TFdel)/mean(WT)):
# gcr2_effect_size
# log(mean(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;])/mean(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# # &quot;log-scale&quot; fold change: mean(log2(TFdel))/mean(log2(WT)):
# mean(log2(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;]))/mean(log2(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# # this is roughly the same as cer
# 
# #### repeat with hybrid to see if LFC/effect sizes match parents
# # hyc
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts_allele$cer[,gene_idx]), redhuis_info_allele$cer) |&gt; 
#   filter(time_point_str == &quot;16 h, low N&quot;)
# genedf$genotype &lt;- as.factor(genedf$genotype) |&gt; relevel(ref = &quot;WT&quot;)
# sigGenotypes &lt;- TFdeldfs_pois$redhuisdf_hyc |&gt; 
#   filter(gene_name == gene_idx &amp; abs(coef) &gt; coef_thresh &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(genedf, aes(x = genotype, y = expr)) + 
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) + 
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# mod &lt;- glm(expr ~ genotype, data = genedf, family = poisson(link = &quot;log&quot;))
# mod$coefficients |&gt; sort(decreasing = TRUE) # a cutoff of 1 would unfortunately still get GCR2, but that might be necessary
# # now tf vs wt for gln3 specifically
# genedf$p_hat &lt;- mod$fitted.values
# gcr2_effect_size &lt;- mod$coefficients[&quot;genotypeGCR2delete&quot;] |&gt; as.numeric()
# slope &lt;- exp(gcr2_effect_size)
# intercept &lt;- mod$coefficients[&quot;(Intercept)&quot;] |&gt; as.numeric()
# plotdf &lt;- pivot_longer(genedf, cols = c(p_hat, expr))
# ggplot(data = plotdf, aes(x = genotype, y = value)) +
#   geom_point(aes(color = name, shape = as.factor(time_point_str))) +
#   scale_x_discrete(breaks = NULL)
# plotdf &lt;- plotdf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
#   pivot_wider(id_cols = c(&quot;time_point_str&quot;, &quot;name&quot;, &quot;well_flask_ID&quot;), 
#               names_from = &quot;genotype&quot;,
#               values_from = &quot;value&quot;)
# ggplot(plotdf, aes(x = WT, y = GCR2delete)) + 
#   geom_point(aes(color = name, shape = time_point_str)) +
#   geom_abline(slope = 1, intercept = 0) +
#   geom_abline(slope = slope, intercept = intercept, color = &quot;gold&quot;) +
#   xlim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE))) +
#   ylim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE)))
# # genedf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
# #   select(expr, p_hat, genotype, well_flask_ID) |&gt; View()
# # mean WT:
# genedf |&gt; filter(genotype == &quot;WT&quot;) |&gt; select(expr) |&gt; pull() |&gt; mean()
# # mean GCR2delete:
# genedf |&gt; filter(genotype == &quot;GCR2delete&quot;) |&gt; select(expr) |&gt; pull() |&gt; mean()
# 
# # effect size, log(mean(TFdel)/mean(WT)):
# gcr2_effect_size
# log(mean(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;])/mean(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# log2(mean(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;])/mean(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# log2(exp(gcr2_effect_size))
# # log fold change: mean(log2(TFdel))/mean(log2(WT)):
# mean(log2(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;]))/mean(log2(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# 
# # repeat with hyp
# genedf &lt;- bind_cols(tibble(expr = redhuis_cts_allele$par[,gene_idx]), redhuis_info_allele$par) |&gt; 
#   filter(time_point_str == &quot;16 h, low N&quot;)
# genedf$genotype &lt;- as.factor(genedf$genotype) |&gt; relevel(ref = &quot;WT&quot;)
# sigGenotypes &lt;- TFdeldfs_pois$redhuisdf_hyp |&gt; 
#   filter(gene_name == gene_idx &amp; abs(coef) &gt; coef_thresh &amp; pval &lt; p_thresh) |&gt; 
#   select(deletion) |&gt; pull()
# ggplot(genedf, aes(x = genotype, y = expr)) + 
#   geom_point(aes(color = gsub(&quot;delete&quot;, &quot;&quot;, genotype) %in% sigGenotypes)) + 
#   theme(axis.text.x = element_text(angle = 90),
#         legend.position = &quot;none&quot;)
# mod &lt;- glm(expr ~ genotype, data = genedf, family = poisson(link = &quot;log&quot;))
# mod$coefficients |&gt; sort(decreasing = TRUE) # a cutoff of 1 would unfortunately still get GCR2, but that might be necessary
# # now tf vs wt for gln3 specifically
# genedf$p_hat &lt;- mod$fitted.values
# gcr2_effect_size &lt;- mod$coefficients[&quot;genotypeGCR2delete&quot;] |&gt; as.numeric()
# slope &lt;- exp(gcr2_effect_size)
# intercept &lt;- mod$coefficients[&quot;(Intercept)&quot;] |&gt; as.numeric()
# plotdf &lt;- pivot_longer(genedf, cols = c(p_hat, expr))
# ggplot(data = plotdf, aes(x = genotype, y = value)) +
#   geom_point(aes(color = name, shape = as.factor(time_point_str))) +
#   scale_x_discrete(breaks = NULL)
# plotdf &lt;- plotdf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
#   pivot_wider(id_cols = c(&quot;time_point_str&quot;, &quot;name&quot;, &quot;well_flask_ID&quot;), 
#               names_from = &quot;genotype&quot;,
#               values_from = &quot;value&quot;)
# ggplot(plotdf, aes(x = WT, y = GCR2delete)) + 
#   geom_point(aes(color = name, shape = time_point_str)) +
#   geom_abline(slope = 1, intercept = 0) +
#   geom_abline(slope = slope, intercept = intercept, color = &quot;gold&quot;) +
#   xlim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE))) +
#   ylim(c(0, max(c(plotdf$WT, plotdf$GCR2delete), na.rm = TRUE)))
# # genedf |&gt; filter(genotype %in% c(&quot;WT&quot;, &quot;GCR2delete&quot;)) |&gt; 
# #   select(expr, p_hat, genotype, well_flask_ID) |&gt; View()
# # mean WT:
# genedf |&gt; filter(genotype == &quot;WT&quot;) |&gt; select(expr) |&gt; pull() |&gt; mean()
# # mean GCR2delete:
# genedf |&gt; filter(genotype == &quot;GCR2delete&quot;) |&gt; select(expr) |&gt; pull() |&gt; mean()
# 
# # effect size, log(mean(TFdel)/mean(WT)):
# gcr2_effect_size
# log(mean(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;])/mean(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# log2(mean(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;])/mean(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# log2(exp(gcr2_effect_size))
# # log fold change: mean(log2(TFdel))/mean(log2(WT)):
# mean(log2(genedf$expr[genedf$genotype == &quot;GCR2delete&quot;]))/mean(log2(genedf$expr[genedf$genotype == &quot;WT&quot;]))
# 
# 
# # secondary goal: find an appropriate threshold for our chosen
# # measure of expr change
# # YNL065W is a good example to explore.
# # Much stronger effect in GLN3 in hyp (somewhat in hyc)
# # than other genotypes, but most genotypes are called sig in glms
# gene_idx &lt;- &quot;YNL065W&quot;
# # hyp has one main regulator: GLN3
# # Then a couple minor ones: AFT1, HAP1, MBP1, ROX1, SOK2
# # Then a few dubious ones we wouldn&#39;t want to count: GCR2 (maybe real but v noisy), then basically all the other TFs except DAL80, PHD1, and TEC1 are called significant but they shouldn&#39;t be
# # TODO: figure out what effect size threshold works for this gene in hyp
# # then see if it&#39;s the same threshold that works for parents (just par for starters)
# 
# # Note: YNL065W no longer seems like a good example --- not 
# # obviously more DE in GLN3delete (it&#39;s really just lowly expressed and noisy)</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
